<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Common batch-effect correction methods imperfectly resolve batch effect • BatchNorm</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Common batch-effect correction methods imperfectly resolve batch effect">
<meta property="og:description" content='How to test the performance of batch effect correction algorithms? We apply three popular batch effect correction workflows to scRNA-Seq libraries from three different donors, with batch effects detailed in Figure 2 and the vignette titled "Sample Donor Effects". We assess the performance of each method using CMS and iLISI scoring. The full results of this test are described in our published Figure 4 and accompanying text.
'>
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BatchNorm</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BatchEffect_Correction.html">Common batch-effect correction methods imperfectly resolve batch effect</a>
    </li>
    <li>
      <a href="../articles/Biaxial_Gating.html">Biaxial Gating of a Single Sample (Fig. S3)</a>
    </li>
    <li>
      <a href="../articles/NormalizationMethods.html">Data normalization and merging strategies differentially impact the measured batch effect</a>
    </li>
    <li>
      <a href="../articles/Pooled_Sequencing.html">Pooling samples for sequencing does not appreciably improve the measured batch effect</a>
    </li>
    <li>
      <a href="../articles/SampleDonor_Effect.html">Batch effects are generated by sample donor</a>
    </li>
    <li>
      <a href="../articles/Sequencing_Effect.html">Batch effects are generated by sequencing depth, but not sequencing replicates alone</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="BatchEffect_Correction_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Common batch-effect correction methods imperfectly resolve batch effect</h1>
                        <h4 class="author">Benjamin R Babcock</h4>
            
      
      
      <div class="hidden name"><code>BatchEffect_Correction.Rmd</code></div>

    </div>

    
    
<div id="harmony-batch-correction" class="section level1">
<h1 class="hasAnchor">
<a href="#harmony-batch-correction" class="anchor"></a>Harmony Batch Correction</h1>
<div id="processing-samples-1-3-seurat-objects" class="section level2">
<h2 class="hasAnchor">
<a href="#processing-samples-1-3-seurat-objects" class="anchor"></a>Processing Samples 1-3 Seurat objects</h2>
<div id="data-filtering-normalization" class="section level3">
<h3 class="hasAnchor">
<a href="#data-filtering-normalization" class="anchor"></a>Data filtering &amp; normalization</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ghosn-lab.github.io/BatchNorm">BatchNorm</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">harmony</span><span class="op">)</span>
<span class="co"># Import unfiltered Seurat object (included with 'BatchNorm' package)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span>

<span class="co"># Run "standard" Seurat + Harmony workflow:</span>
<span class="co">## "htmlpreview.github.io/?https://github.com/immunogenomics/harmony/blob/master/docs/SeuratV3.html"</span>
<span class="co"># Including filtering by mitochondrial percentage (+5 SD)</span>
<span class="co"># Including data normalization, variable gene selection and gene scaling (performed on all samples together)</span>
<span class="va">PBMCs</span> <span class="op">&lt;-</span> <span class="va">PBMCs</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span> 
         <span class="fu"><a href="../reference/MitoFilter.html">MitoFilter</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span> 
         <span class="fu">NormalizeData</span><span class="op">(</span>normalization.method <span class="op">=</span> <span class="st">"LogNormalize"</span>, assay <span class="op">=</span> <span class="st">"RNA"</span>, scale.factor <span class="op">=</span> <span class="fl">10000</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span> 
         <span class="fu">NormalizeData</span><span class="op">(</span>verbose <span class="op">=</span> <span class="cn">FALSE</span>, assay <span class="op">=</span> <span class="st">"ADT"</span>, normalization.method <span class="op">=</span> <span class="st">"CLR"</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span>
         <span class="fu">FindVariableFeatures</span><span class="op">(</span>selection.method <span class="op">=</span> <span class="st">"vst"</span>, nfeatures <span class="op">=</span> <span class="fl">2000</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span>
         <span class="fu">ScaleData</span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span> 
         <span class="fu">RunPCA</span><span class="op">(</span>npcs <span class="op">=</span> <span class="fl">30</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span> 
         <span class="fu"><a href="https://rdrr.io/pkg/harmony/man/RunHarmony.html">RunHarmony</a></span><span class="op">(</span><span class="st">"orig.ident"</span>, plot_convergence <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="BatchEffect_Correction_files/figure-html/inital_processing_harmony-1.png" width="672" style="display: block; margin: auto;"></p>
</div>
</div>
<div id="generating-a-umap-and-clusters" class="section level2">
<h2 class="hasAnchor">
<a href="#generating-a-umap-and-clusters" class="anchor"></a>Generating a UMAP and clusters</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Generating a UMAP and clusters</span>
<span class="va">PBMCs</span> <span class="op">&lt;-</span> <span class="va">PBMCs</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span> 
         <span class="fu">RunUMAP</span><span class="op">(</span>reduction <span class="op">=</span> <span class="st">"harmony"</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">14</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span>
         <span class="fu">FindNeighbors</span><span class="op">(</span>reduction <span class="op">=</span> <span class="st">"harmony"</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">14</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span>
         <span class="fu">FindClusters</span><span class="op">(</span>resolution <span class="op">=</span> <span class="fl">.8</span><span class="op">)</span></code></pre></div>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 16946
## Number of edges: 588809
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.8822
## Number of communities: 20
## Elapsed time: 2 seconds</code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">UMAPPlot</span><span class="op">(</span><span class="va">PBMCs</span>, cols <span class="op">=</span> <span class="va">colors.use</span>, group.by <span class="op">=</span> <span class="st">"orig.ident"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Harmony: Sample Identity"</span><span class="op">)</span></code></pre></div>
<p><img src="BatchEffect_Correction_files/figure-html/Clustering_harmony-1.png" width="672" style="display: block; margin: auto;"></p>
<div id="measuring-sample-umap-integration-generating-an-ilisi-score" class="section level3">
<h3 class="hasAnchor">
<a href="#measuring-sample-umap-integration-generating-an-ilisi-score" class="anchor"></a>Measuring sample-UMAP integration (generating an iLISI score)</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/GetiLISI.html">GetiLISI</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMCs</span>, nSamples <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0.1828125</code></pre>
</div>
</div>
<div id="cell-typing-of-harmonized-pbmcs-object-for-cms" class="section level2">
<h2 class="hasAnchor">
<a href="#cell-typing-of-harmonized-pbmcs-object-for-cms" class="anchor"></a>Cell Typing of “harmonized” PBMCs object for CMS</h2>
<div id="convert-cluster-classifications-to-cell-type-classifications" class="section level3">
<h3 class="hasAnchor">
<a href="#convert-cluster-classifications-to-cell-type-classifications" class="anchor"></a>Convert cluster classifications to cell type classifications</h3>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># For complete cell classification workflow see our vignette "Biaxial Gating of a Single Sample"</span>
<span class="co"># More details can be found in figure S3 of our manuscript "Data Matrix Normalization and Merging Strategies Minimize Batch-specific Systemic Variation in scRNA-Seq Data."</span>
<span class="fu">UMAPPlot</span><span class="op">(</span><span class="va">PBMCs</span>, cols <span class="op">=</span> <span class="va">colors.use</span>, pt.size <span class="op">=</span> <span class="fl">2</span>, 
         group.by <span class="op">=</span> <span class="st">"seurat_clusters"</span>, label <span class="op">=</span> <span class="cn">T</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Harmony: Clusters"</span><span class="op">)</span></code></pre></div>
<p><img src="BatchEffect_Correction_files/figure-html/Cluster_assignment_harmony-1.png" width="672" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># B_Cells = 10, 11</span>
<span class="co"># T_CD4 = 0, 1, 13, 16</span>
<span class="co"># No TReg cluster (contained within cluster 1)</span>
<span class="co"># T_CD8 = 7, 2</span>
<span class="co"># NK_T = 5</span>
<span class="co"># NK = 3, 18</span>
<span class="co"># NKCD56Hi = 9</span>
<span class="co"># Monocyte_Classical = 4, 6, 12</span>
<span class="co"># Monocyte_NonClassical = 8</span>
<span class="co"># Dendritic_Cells = 14, 15</span>
<span class="co"># HSPCs = 19</span>
<span class="co"># Cycling_Cells = 17</span>
<span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">PBMCs</span><span class="op">[[</span><span class="st">"seurat_clusters"</span><span class="op">]</span><span class="op">]</span>
<span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">plyr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyr/man/mapvalues.html">mapvalues</a></span><span class="op">(</span><span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span>, from <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">11</span>, <span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">13</span>, <span class="fl">16</span>,
                                                         <span class="fl">7</span>, <span class="fl">2</span>, <span class="fl">5</span>, <span class="fl">3</span>, <span class="fl">18</span>, <span class="fl">9</span>,
                                                         <span class="fl">4</span>, <span class="fl">6</span>, <span class="fl">12</span>, 
                                                         <span class="fl">8</span>, <span class="fl">14</span>, <span class="fl">15</span>,
                                                         <span class="fl">19</span>, <span class="fl">17</span><span class="op">)</span>, 
                to <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'B_Cells'</span>, <span class="st">'B_Cells'</span>, <span class="st">'T_CD4'</span>, <span class="st">'T_CD4'</span>, <span class="st">'T_CD4'</span>, <span class="st">'T_CD4'</span>,
                       <span class="st">'T_CD8'</span>, <span class="st">'T_CD8'</span>, <span class="st">'NK_T'</span>, <span class="st">'NK'</span>, <span class="st">"NK"</span>, <span class="st">"NK_CD56Hi"</span>,
                       <span class="st">'Monocyte_Classical'</span>, <span class="st">'Monocyte_Classical'</span>, <span class="st">'Monocyte_Classical'</span>,
                       <span class="st">'Monocyte_NonClassical'</span>, <span class="st">'Dendritic_Cells'</span>, <span class="st">'Dendritic_Cells'</span>,
                       <span class="st">'HSPCs'</span>, <span class="st">'Cycling_Cells'</span><span class="op">)</span><span class="op">)</span>
<span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span>,
                                 levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"B_Cells"</span>, <span class="st">"T_CD4"</span>, <span class="st">"TReg"</span>, 
                                            <span class="st">"T_CD8"</span>, <span class="st">"NK_T"</span>, <span class="st">"NK"</span>, <span class="st">"NK_CD56Hi"</span>,
                                            <span class="st">"Monocyte_Classical"</span>, <span class="st">"Monocyte_NonClassical"</span>,
                                            <span class="st">"Dendritic_Cells"</span>, <span class="st">"HSPCs"</span>, <span class="st">"Cycling_Cells"</span><span class="op">)</span><span class="op">)</span>
<span class="va">PBMCs</span><span class="op">[[</span><span class="st">"Cell_Type"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span>
<span class="fu">UMAPPlot</span><span class="op">(</span><span class="va">PBMCs</span>, cols <span class="op">=</span> <span class="va">colors.use</span>, label <span class="op">=</span> <span class="cn">F</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Harmony: Cell Type Classifications"</span><span class="op">)</span></code></pre></div>
<p><img src="BatchEffect_Correction_files/figure-html/Cluster_assignment_harmony-2.png" width="672" style="display: block; margin: auto;"></p>
</div>
</div>
<div id="compare-single-sample-workflow-cell-type-classifications-to-joint-classifications-generating-a-cms" class="section level2">
<h2 class="hasAnchor">
<a href="#compare-single-sample-workflow-cell-type-classifications-to-joint-classifications-generating-a-cms" class="anchor"></a>Compare single-sample workflow cell type classifications to joint classifications (generating a CMS)</h2>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># PBMC Sample 1</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">PBMC1_Single_ID</span><span class="op">)</span>
<span class="va">S1cms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetCMS.html">GetCMS</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMCs</span>, 
                sample.ID <span class="op">=</span> <span class="st">"Sample_01"</span>, 
                reference.ID <span class="op">=</span> <span class="va">PBMC1_Single_ID</span><span class="op">)</span>
<span class="co"># PBMC Sample 2</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">PBMC2_Single_ID</span><span class="op">)</span>
<span class="va">S2cms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetCMS.html">GetCMS</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMCs</span>, 
                sample.ID <span class="op">=</span> <span class="st">"Sample_02"</span>, 
                reference.ID <span class="op">=</span> <span class="va">PBMC2_Single_ID</span><span class="op">)</span>
<span class="co"># PBMC Sample 3</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">PBMC3_Single_ID</span><span class="op">)</span>
<span class="va">S3cms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetCMS.html">GetCMS</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMCs</span>,
                sample.ID <span class="op">=</span> <span class="st">"Sample_03"</span>,
                reference.ID <span class="op">=</span> <span class="va">PBMC3_Single_ID</span><span class="op">)</span>

<span class="co"># Average CMS</span>
<span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">S1cms</span>, <span class="va">S2cms</span>, <span class="va">S3cms</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0.1534497</code></pre>
</div>
</div>
<div id="liger-batch-correction" class="section level1">
<h1 class="hasAnchor">
<a href="#liger-batch-correction" class="anchor"></a>LIGER Batch Correction</h1>
<div id="processing-samples-1-3-seurat-objects-1" class="section level2">
<h2 class="hasAnchor">
<a href="#processing-samples-1-3-seurat-objects-1" class="anchor"></a>Processing Samples 1-3 Seurat objects</h2>
<div id="data-filtering-normalization-1" class="section level3">
<h3 class="hasAnchor">
<a href="#data-filtering-normalization-1" class="anchor"></a>Data filtering &amp; normalization</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/welch-lab/liger">rliger</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">SeuratWrappers</span><span class="op">)</span>

<span class="co"># Import unfiltered Seurat object (included with 'BatchNorm' package)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span>

<span class="co"># Run "standard" Seurat + LIGER workflow:</span>
<span class="co">## "https://github.com/satijalab/seurat-wrappers/blob/master/docs/liger.md"</span>
<span class="co"># Including filtering by mitochondrial percentage (+5 SD)</span>
<span class="co"># Including data normalization, variable gene selection and gene scaling (performed on all samples together)</span>
<span class="va">PBMCs</span> <span class="op">&lt;-</span> <span class="va">PBMCs</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span> 
         <span class="fu"><a href="../reference/MitoFilter.html">MitoFilter</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span> 
         <span class="fu">NormalizeData</span><span class="op">(</span>normalization.method <span class="op">=</span> <span class="st">"LogNormalize"</span>, assay <span class="op">=</span> <span class="st">"RNA"</span>, scale.factor <span class="op">=</span> <span class="fl">10000</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span> 
         <span class="fu">NormalizeData</span><span class="op">(</span>verbose <span class="op">=</span> <span class="cn">FALSE</span>, assay <span class="op">=</span> <span class="st">"ADT"</span>, normalization.method <span class="op">=</span> <span class="st">"CLR"</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span>
         <span class="fu">FindVariableFeatures</span><span class="op">(</span>selection.method <span class="op">=</span> <span class="st">"vst"</span>, nfeatures <span class="op">=</span> <span class="fl">2000</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span>
         <span class="fu">ScaleData</span><span class="op">(</span>split.by <span class="op">=</span> <span class="st">"orig.ident"</span>, do.center <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span> 
         <span class="fu"><a href="https://rdrr.io/pkg/SeuratWrappers/man/RunOptimizeALS.html">RunOptimizeALS</a></span><span class="op">(</span>k <span class="op">=</span> <span class="fl">20</span>, lambda <span class="op">=</span> <span class="fl">5</span>, split.by <span class="op">=</span> <span class="st">"orig.ident"</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span> 
         <span class="fu"><a href="https://rdrr.io/pkg/SeuratWrappers/man/RunQuantileNorm.html">RunQuantileNorm</a></span><span class="op">(</span>split.by <span class="op">=</span> <span class="st">"orig.ident"</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span>
         <span class="fu">RunUMAP</span><span class="op">(</span>dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">14</span>, reduction <span class="op">=</span> <span class="st">"iNMF"</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span> 
         <span class="fu">FindNeighbors</span><span class="op">(</span>reduction <span class="op">=</span> <span class="st">"iNMF"</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">14</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span>
         <span class="fu">FindClusters</span><span class="op">(</span>resolution <span class="op">=</span> <span class="fl">0.8</span><span class="op">)</span></code></pre></div>
<pre><code>## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |==                                                                    |   3%
  |                                                                            
  |=====                                                                 |   7%
  |                                                                            
  |=======                                                               |  10%
  |                                                                            
  |=========                                                             |  13%
  |                                                                            
  |============                                                          |  17%
  |                                                                            
  |==============                                                        |  20%
  |                                                                            
  |================                                                      |  23%
  |                                                                            
  |===================                                                   |  27%
  |                                                                            
  |=====================                                                 |  30%
  |                                                                            
  |=======================                                               |  33%
  |                                                                            
  |==========================                                            |  37%
  |                                                                            
  |============================                                          |  40%
  |                                                                            
  |==============================                                        |  43%
  |                                                                            
  |=================================                                     |  47%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |=====================================                                 |  53%
  |                                                                            
  |========================================                              |  57%
  |                                                                            
  |==========================================                            |  60%
  |                                                                            
  |============================================                          |  63%
  |                                                                            
  |===============================================                       |  67%
  |                                                                            
  |=================================================                     |  70%
  |                                                                            
  |===================================================                   |  73%
  |                                                                            
  |======================================================                |  77%
  |                                                                            
  |========================================================              |  80%
  |                                                                            
  |==========================================================            |  83%
  |                                                                            
  |=============================================================         |  87%
  |                                                                            
  |===============================================================       |  90%
  |                                                                            
  |=================================================================     |  93%
  |                                                                            
  |====================================================================  |  97%
  |                                                                            
  |======================================================================| 100%
## Finished in 5.541296 mins, 30 iterations.
## Max iterations set: 30.
## Final objective delta: 1.118738e-05.
## Best results with seed 1.
## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 16946
## Number of edges: 534569
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.8882
## Number of communities: 24
## Elapsed time: 1 seconds</code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">UMAPPlot</span><span class="op">(</span><span class="va">PBMCs</span>, cols <span class="op">=</span> <span class="va">colors.use</span>, group.by <span class="op">=</span> <span class="st">"orig.ident"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"LIGER: Sample Identity"</span><span class="op">)</span></code></pre></div>
<p><img src="BatchEffect_Correction_files/figure-html/inital_processing_LIGER-1.png" width="672" style="display: block; margin: auto;"></p>
</div>
</div>
<div id="measuring-sample-umap-integration-generating-an-ilisi-score-1" class="section level2">
<h2 class="hasAnchor">
<a href="#measuring-sample-umap-integration-generating-an-ilisi-score-1" class="anchor"></a>Measuring sample-UMAP integration (generating an iLISI score)</h2>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/GetiLISI.html">GetiLISI</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMCs</span>, nSamples <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0.13214</code></pre>
</div>
<div id="cell-typing-of-liger-corrected-pbmcs-object-for-cms" class="section level2">
<h2 class="hasAnchor">
<a href="#cell-typing-of-liger-corrected-pbmcs-object-for-cms" class="anchor"></a>Cell Typing of LIGER-corrected PBMCs object for CMS</h2>
<div id="convert-cluster-classifications-to-cell-type-classifications-1" class="section level3">
<h3 class="hasAnchor">
<a href="#convert-cluster-classifications-to-cell-type-classifications-1" class="anchor"></a>Convert cluster classifications to cell type classifications</h3>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># For complete cell classification workflow see our vignette "Biaxial Gating of a Single Sample"</span>
<span class="co"># More details can be found in figure S3 of our manuscript "Data Matrix Normalization and Merging Strategies Minimize Batch-specific Systemic Variation in scRNA-Seq Data."</span>
<span class="fu">UMAPPlot</span><span class="op">(</span><span class="va">PBMCs</span>, cols <span class="op">=</span> <span class="va">colors.use</span>, pt.size <span class="op">=</span> <span class="fl">2</span>, 
         group.by <span class="op">=</span> <span class="st">"clusters"</span>, label <span class="op">=</span> <span class="cn">T</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"LIGER: Clusters"</span><span class="op">)</span></code></pre></div>
<p><img src="BatchEffect_Correction_files/figure-html/Cluster_assignment_LIGER-1.png" width="672" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># B_Cells = 3, 10, 12, 17</span>
<span class="co"># T_CD4 = 2, 19</span>
<span class="co"># No TReg cluster (contained within cluster 2)</span>
<span class="co"># T_CD8 = 20, 15</span>
<span class="co"># NK_T = 4</span>
<span class="co"># NK = 6, 16</span>
<span class="co"># NK_CD56Hi = 8</span>
<span class="co"># Monocyte_Classical = 1, 7, 9, 14</span>
<span class="co"># Monocyte_NonClassical = 18</span>
<span class="co"># Dendritic_Cells = 11, 13</span>
<span class="co"># HSPCs = 5</span>
<span class="co"># Cycling_Cells no cluster</span>

<span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">PBMCs</span><span class="op">[[</span><span class="st">"clusters"</span><span class="op">]</span><span class="op">]</span>
<span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">plyr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyr/man/mapvalues.html">mapvalues</a></span><span class="op">(</span><span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span>, from <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">10</span>, <span class="fl">12</span>, <span class="fl">17</span>, <span class="fl">2</span>, <span class="fl">19</span>,
                                                         <span class="fl">20</span>, <span class="fl">15</span>, <span class="fl">4</span>, <span class="fl">6</span>, <span class="fl">16</span>, <span class="fl">8</span>,
                                                         <span class="fl">1</span>, <span class="fl">7</span>, <span class="fl">9</span>, 
                                                         <span class="fl">14</span>, <span class="fl">18</span>, <span class="fl">11</span>,
                                                         <span class="fl">13</span>, <span class="fl">5</span><span class="op">)</span>, 
                to <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'B_Cells'</span>, <span class="st">'B_Cells'</span>, <span class="st">'B_Cells'</span>, <span class="st">'B_Cells'</span>, <span class="st">'T_CD4'</span>, <span class="st">'T_CD4'</span>,
                       <span class="st">'T_CD8'</span>, <span class="st">'T_CD8'</span>, <span class="st">'NK_T'</span>, <span class="st">'NK'</span>, <span class="st">"NK"</span>, <span class="st">"NK_CD56Hi"</span>,
                       <span class="st">'Monocyte_Classical'</span>, <span class="st">'Monocyte_Classical'</span>, <span class="st">'Monocyte_Classical'</span>,
                       <span class="st">'Monocyte_Classical'</span>, <span class="st">'Monocyte_NonClassical'</span>, <span class="st">'Dendritic_Cells'</span>,
                       <span class="st">'Dendritic_Cells'</span>, <span class="st">'HSPCs'</span><span class="op">)</span><span class="op">)</span>
<span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span>,
                                 levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"B_Cells"</span>, <span class="st">"T_CD4"</span>, <span class="st">"TReg"</span>, 
                                            <span class="st">"T_CD8"</span>, <span class="st">"NK_T"</span>, <span class="st">"NK"</span>, <span class="st">"NK_CD56Hi"</span>,
                                            <span class="st">"Monocyte_Classical"</span>, <span class="st">"Monocyte_NonClassical"</span>,
                                            <span class="st">"Dendritic_Cells"</span>, <span class="st">"HSPCs"</span>, <span class="st">"Cycling_Cells"</span><span class="op">)</span><span class="op">)</span>
<span class="va">PBMCs</span><span class="op">[[</span><span class="st">"Cell_Type"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span>
<span class="fu">UMAPPlot</span><span class="op">(</span><span class="va">PBMCs</span>, cols <span class="op">=</span> <span class="va">colors.use</span>, label <span class="op">=</span> <span class="cn">F</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"LIGER: Cell Type Classifications"</span><span class="op">)</span></code></pre></div>
<p><img src="BatchEffect_Correction_files/figure-html/Cluster_assignment_LIGER-2.png" width="672" style="display: block; margin: auto;"></p>
</div>
</div>
<div id="compare-single-sample-workflow-cell-type-classifications-to-joint-classifications-generating-a-cms-1" class="section level2">
<h2 class="hasAnchor">
<a href="#compare-single-sample-workflow-cell-type-classifications-to-joint-classifications-generating-a-cms-1" class="anchor"></a>Compare single-sample workflow cell type classifications to joint classifications (generating a CMS)</h2>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># PBMC Sample 1</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">PBMC1_Single_ID</span><span class="op">)</span>
<span class="va">S1cms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetCMS.html">GetCMS</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMCs</span>, 
                sample.ID <span class="op">=</span> <span class="st">"Sample_01"</span>, 
                reference.ID <span class="op">=</span> <span class="va">PBMC1_Single_ID</span><span class="op">)</span>
<span class="co"># PBMC Sample 2</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">PBMC2_Single_ID</span><span class="op">)</span>
<span class="va">S2cms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetCMS.html">GetCMS</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMCs</span>, 
                sample.ID <span class="op">=</span> <span class="st">"Sample_02"</span>, 
                reference.ID <span class="op">=</span> <span class="va">PBMC2_Single_ID</span><span class="op">)</span>
<span class="co"># PBMC Sample 3</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">PBMC3_Single_ID</span><span class="op">)</span>
<span class="va">S3cms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetCMS.html">GetCMS</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMCs</span>,
                sample.ID <span class="op">=</span> <span class="st">"Sample_03"</span>,
                reference.ID <span class="op">=</span> <span class="va">PBMC3_Single_ID</span><span class="op">)</span>

<span class="co"># Average CMS</span>
<span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">S1cms</span>, <span class="va">S2cms</span>, <span class="va">S3cms</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0.2572457</code></pre>
</div>
</div>
<div id="seurat-3-batch-correction" class="section level1">
<h1 class="hasAnchor">
<a href="#seurat-3-batch-correction" class="anchor"></a>Seurat 3 Batch Correction</h1>
<div id="processing-samples-1-3-seurat-objects-2" class="section level2">
<h2 class="hasAnchor">
<a href="#processing-samples-1-3-seurat-objects-2" class="anchor"></a>Processing Samples 1-3 Seurat objects</h2>
<div id="data-filtering-normalization-2" class="section level3">
<h3 class="hasAnchor">
<a href="#data-filtering-normalization-2" class="anchor"></a>Data filtering &amp; normalization</h3>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Import unfiltered Seurat object (included with 'BatchNorm' package)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span>

<span class="co"># Run "standard" Seurat 3 data integration workflow:</span>
<span class="co"># "https://satijalab.org/seurat/archive/v3.0/integration.html"</span>
<span class="co"># Including filtering by mitochondrial percentage (+5 SD)</span>
<span class="va">PBMCs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MitoFilter.html">MitoFilter</a></span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span>
<span class="co"># Including SCTransform data normalization and feature selection</span>
<span class="va">PBMCs.list</span> <span class="op">&lt;-</span> <span class="fu">SplitObject</span><span class="op">(</span><span class="va">PBMCs</span>, split.by <span class="op">=</span> <span class="st">"orig.ident"</span><span class="op">)</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">PBMCs.list</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">PBMCs.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">SCTransform</span><span class="op">(</span><span class="va">PBMCs.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="op">}</span>
<span class="co"># Next, SelectIntegrationFeatures identifies a given number of features to use during the integration steps (here, 3,000).</span>
<span class="co"># PrepSCTIntegration ensures all pearson residuals have been calculated and that the data is ready for integration.</span>
<span class="va">PBMCs.features</span> <span class="op">&lt;-</span> <span class="fu">SelectIntegrationFeatures</span><span class="op">(</span>object.list <span class="op">=</span> <span class="va">PBMCs.list</span>, nfeatures <span class="op">=</span> <span class="fl">3000</span><span class="op">)</span>
<span class="co"># Set global object size to 2 gb to avoid errors in this next step.</span>
<span class="fu"><a href="https://rdrr.io/r/base/options.html">options</a></span><span class="op">(</span>future.globals.maxSize <span class="op">=</span> <span class="fl">2000</span> <span class="op">*</span> <span class="fl">1024</span><span class="op">^</span><span class="fl">2</span><span class="op">)</span>
<span class="va">PBMCs.list</span> <span class="op">&lt;-</span> <span class="fu">PrepSCTIntegration</span><span class="op">(</span>object.list <span class="op">=</span> <span class="va">PBMCs.list</span>, anchor.features <span class="op">=</span> <span class="va">PBMCs.features</span>,
                                 verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="co"># Identify integration "anchors" and integrate the datasets.</span>
<span class="co"># This may take approximately 10 minutes or more on this data set</span>
<span class="va">PBMCs.anchors</span> <span class="op">&lt;-</span> <span class="fu">FindIntegrationAnchors</span><span class="op">(</span>object.list <span class="op">=</span> <span class="va">PBMCs.list</span>, 
                                        normalization.method <span class="op">=</span> <span class="st">"SCT"</span>, 
                                        anchor.features <span class="op">=</span> <span class="va">PBMCs.features</span>, 
                                        verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">PBMCs</span> <span class="op">&lt;-</span> <span class="fu">IntegrateData</span><span class="op">(</span>anchorset <span class="op">=</span> <span class="va">PBMCs.anchors</span>, 
                       normalization.method <span class="op">=</span> <span class="st">"SCT"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="co"># Next, we proceed with a standard UMAP/clustering/cell-typing analysis on the integrated dataset</span>
<span class="co"># We are careful not to run the ScaleData function after integration (which would override the integrated features).</span>
<span class="va">PBMCs</span> <span class="op">&lt;-</span> <span class="fu">RunPCA</span><span class="op">(</span><span class="va">PBMCs</span>, npcs <span class="op">=</span> <span class="fl">30</span><span class="op">)</span>
<span class="va">PBMCs</span> <span class="op">&lt;-</span> <span class="va">PBMCs</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span> 
         <span class="fu">RunUMAP</span><span class="op">(</span>reduction <span class="op">=</span> <span class="st">"pca"</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">14</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span>
         <span class="fu">FindNeighbors</span><span class="op">(</span>reduction <span class="op">=</span> <span class="st">"pca"</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">14</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span>
         <span class="fu">FindClusters</span><span class="op">(</span>resolution <span class="op">=</span> <span class="fl">.8</span><span class="op">)</span></code></pre></div>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 16946
## Number of edges: 606600
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.8978
## Number of communities: 19
## Elapsed time: 3 seconds</code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">PBMCs</span> <span class="op">&lt;-</span> <span class="fu">NormalizeData</span><span class="op">(</span><span class="va">PBMCs</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span>, assay <span class="op">=</span> <span class="st">"ADT"</span>, normalization.method <span class="op">=</span> <span class="st">"CLR"</span><span class="op">)</span>

<span class="fu">UMAPPlot</span><span class="op">(</span><span class="va">PBMCs</span>, cols <span class="op">=</span> <span class="va">colors.use</span>, group.by <span class="op">=</span> <span class="st">"orig.ident"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Seurat 3: Sample Identity"</span><span class="op">)</span></code></pre></div>
<p><img src="BatchEffect_Correction_files/figure-html/inital_processing_seurat3-1.png" width="672" style="display: block; margin: auto;"></p>
</div>
</div>
<div id="measuring-sample-umap-integration-generating-an-ilisi-score-2" class="section level2">
<h2 class="hasAnchor">
<a href="#measuring-sample-umap-integration-generating-an-ilisi-score-2" class="anchor"></a>Measuring sample-UMAP integration (generating an iLISI score)</h2>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/GetiLISI.html">GetiLISI</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMCs</span>, nSamples <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0.1913981</code></pre>
</div>
<div id="cell-typing-of-seurat-3-corrected-pbmcs-object-for-cms" class="section level2">
<h2 class="hasAnchor">
<a href="#cell-typing-of-seurat-3-corrected-pbmcs-object-for-cms" class="anchor"></a>Cell Typing of Seurat 3-corrected PBMCs object for CMS</h2>
<div id="convert-cluster-classifications-to-cell-type-classifications-2" class="section level3">
<h3 class="hasAnchor">
<a href="#convert-cluster-classifications-to-cell-type-classifications-2" class="anchor"></a>Convert cluster classifications to cell type classifications</h3>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># For complete cell classification workflow see our vignette "Biaxial Gating of a Single Sample"</span>
<span class="co"># More details can be found in figure S3 of our manuscript "Data Matrix Normalization and Merging Strategies Minimize Batch-specific Systemic Variation in scRNA-Seq Data."</span>
<span class="fu">UMAPPlot</span><span class="op">(</span><span class="va">PBMCs</span>, cols <span class="op">=</span> <span class="va">colors.use</span>, pt.size <span class="op">=</span> <span class="fl">2</span>, 
         group.by <span class="op">=</span> <span class="st">"seurat_clusters"</span>, label <span class="op">=</span> <span class="cn">T</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Seurat 3: Clusters"</span><span class="op">)</span></code></pre></div>
<p><img src="BatchEffect_Correction_files/figure-html/Cluster_assignment_seurat3-1.png" width="672" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># B_Cells = 11, 14, 16</span>
<span class="co"># T_CD4 = 0, 2, 3, 15</span>
<span class="co"># TReg = 13</span>
<span class="co"># T_CD8 = 5, 8</span>
<span class="co"># NK_T = 7</span>
<span class="co"># NK = 1</span>
<span class="co"># NK_CD56Hi = 12</span>
<span class="co"># Monocyte_Classical = 4, 6, 9</span>
<span class="co"># Monocyte_NonClassical = 10</span>
<span class="co"># Dendritic_Cells = 17</span>
<span class="co"># HSPCs = 18</span>
<span class="co"># Cycling_Cells no cluster</span>
<span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">PBMCs</span><span class="op">[[</span><span class="st">"seurat_clusters"</span><span class="op">]</span><span class="op">]</span>
<span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">plyr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyr/man/mapvalues.html">mapvalues</a></span><span class="op">(</span><span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span>, from <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">11</span>, <span class="fl">14</span>, <span class="fl">16</span>, <span class="fl">0</span>, <span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">15</span>,
                                                         <span class="fl">13</span>, <span class="fl">5</span>, <span class="fl">8</span>, <span class="fl">7</span>, <span class="fl">1</span>, <span class="fl">12</span>,
                                                         <span class="fl">4</span>, <span class="fl">6</span>, <span class="fl">9</span>, 
                                                         <span class="fl">10</span>, <span class="fl">17</span>, <span class="fl">18</span><span class="op">)</span>, 
                to <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'B_Cells'</span>, <span class="st">'B_Cells'</span>, <span class="st">'B_Cells'</span>, <span class="st">'T_CD4'</span>, <span class="st">'T_CD4'</span>, <span class="st">'T_CD4'</span>, <span class="st">'T_CD4'</span>,
                       <span class="st">'TReg'</span>, <span class="st">'T_CD8'</span>, <span class="st">'T_CD8'</span>, <span class="st">'NK_T'</span>, <span class="st">"NK"</span>, <span class="st">"NK_CD56Hi"</span>,
                       <span class="st">'Monocyte_Classical'</span>, <span class="st">'Monocyte_Classical'</span>, <span class="st">'Monocyte_Classical'</span>,
                       <span class="st">'Monocyte_NonClassical'</span>, <span class="st">'Dendritic_Cells'</span>, <span class="st">'HSPCs'</span><span class="op">)</span><span class="op">)</span>
<span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span>,
                                 levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"B_Cells"</span>, <span class="st">"T_CD4"</span>, <span class="st">"TReg"</span>, 
                                            <span class="st">"T_CD8"</span>, <span class="st">"NK_T"</span>, <span class="st">"NK"</span>, <span class="st">"NK_CD56Hi"</span>,
                                            <span class="st">"Monocyte_Classical"</span>, <span class="st">"Monocyte_NonClassical"</span>,
                                            <span class="st">"Dendritic_Cells"</span>, <span class="st">"HSPCs"</span>, <span class="st">"Cycling_Cells"</span><span class="op">)</span><span class="op">)</span>
<span class="va">PBMCs</span><span class="op">[[</span><span class="st">"Cell_Type"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span>
<span class="fu">UMAPPlot</span><span class="op">(</span><span class="va">PBMCs</span>, cols <span class="op">=</span> <span class="va">colors.use</span>, label <span class="op">=</span> <span class="cn">F</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Seurat 3: Cell Type Classifications"</span><span class="op">)</span></code></pre></div>
<p><img src="BatchEffect_Correction_files/figure-html/Cluster_assignment_seurat3-2.png" width="672" style="display: block; margin: auto;"></p>
</div>
</div>
<div id="compare-single-sample-workflow-cell-type-classifications-to-joint-classifications-generating-a-cms-2" class="section level2">
<h2 class="hasAnchor">
<a href="#compare-single-sample-workflow-cell-type-classifications-to-joint-classifications-generating-a-cms-2" class="anchor"></a>Compare single-sample workflow cell type classifications to joint classifications (generating a CMS)</h2>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># PBMC Sample 1</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">PBMC1_Single_ID</span><span class="op">)</span>
<span class="va">S1cms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetCMS.html">GetCMS</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMCs</span>, 
                sample.ID <span class="op">=</span> <span class="st">"Sample_01"</span>, 
                reference.ID <span class="op">=</span> <span class="va">PBMC1_Single_ID</span><span class="op">)</span>
<span class="co"># PBMC Sample 2</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">PBMC2_Single_ID</span><span class="op">)</span>
<span class="va">S2cms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetCMS.html">GetCMS</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMCs</span>, 
                sample.ID <span class="op">=</span> <span class="st">"Sample_02"</span>, 
                reference.ID <span class="op">=</span> <span class="va">PBMC2_Single_ID</span><span class="op">)</span>
<span class="co"># PBMC Sample 3</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">PBMC3_Single_ID</span><span class="op">)</span>
<span class="va">S3cms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetCMS.html">GetCMS</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMCs</span>,
                sample.ID <span class="op">=</span> <span class="st">"Sample_03"</span>,
                reference.ID <span class="op">=</span> <span class="va">PBMC3_Single_ID</span><span class="op">)</span>

<span class="co"># Average CMS</span>
<span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">S1cms</span>, <span class="va">S2cms</span>, <span class="va">S3cms</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0.1816485</code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Benjamin Babcock.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
