<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Data normalization and merging strategies differentially impact the measured batch effect â€¢ BatchNorm</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Data normalization and merging strategies differentially impact the measured batch effect">
<meta property="og:description" content="Data normalization and merging strategies differentially impact the measured batch effect? We measure the CMS and iLISI scores resulting from two data normalization/scaling methods (log-normalization + scaling vs. SCTransform). We perform each method twice - either on individual samples (prior to merging) or on a joint object of all samples (after merging). The results of this analysis are presented in greater detail in Figure 5 of our manuscript.
">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BatchNorm</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BatchEffect_Correction.html">Common batch-effect correction methods imperfectly resolve batch effect</a>
    </li>
    <li>
      <a href="../articles/Biaxial_Gating.html">Biaxial Gating of a Single Sample (Fig. S3)</a>
    </li>
    <li>
      <a href="../articles/NormalizationMethods.html">Data normalization and merging strategies differentially impact the measured batch effect</a>
    </li>
    <li>
      <a href="../articles/Pooled_Sequencing.html">Pooling samples for sequencing does not appreciably improve the measured batch effect</a>
    </li>
    <li>
      <a href="../articles/SampleDonor_Effect.html">Batch effects are generated by sample donor</a>
    </li>
    <li>
      <a href="../articles/Sequencing_Effect.html">Batch effects are generated by sequencing depth, but not sequencing replicates alone</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="NormalizationMethods_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Data normalization and merging strategies differentially impact the measured batch effect</h1>
                        <h4 class="author">Benjamin R Babcock</h4>
            
      
      
      <div class="hidden name"><code>NormalizationMethods.Rmd</code></div>

    </div>

    
    
<div id="log-normalize-scaling-figure-5a" class="section level1">
<h1 class="hasAnchor">
<a href="#log-normalize-scaling-figure-5a" class="anchor"></a>Log-Normalize + Scaling (Figure 5A)</h1>
<div id="prior-to-merging-individual-samples" class="section level2">
<h2 class="hasAnchor">
<a href="#prior-to-merging-individual-samples" class="anchor"></a>Prior to merging (individual samples)</h2>
<div id="data-filtering-normalization" class="section level3">
<h3 class="hasAnchor">
<a href="#data-filtering-normalization" class="anchor"></a>Data filtering &amp; normalization</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ghosn-lab.github.io/BatchNorm">BatchNorm</a></span><span class="op">)</span>
<span class="co"># Import unfiltered Seurat object (included with 'BatchNorm' package)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span>

<span class="co"># Filter sample by mitochondrial percentage (+5 SD)</span>
<span class="va">PBMCs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MitoFilter.html">MitoFilter</a></span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span>
<span class="co"># Run "standard" Seurat workflow on a list of samples, independently:</span>
<span class="co"># Including data normalization, variable gene selection and gene scaling (performed on each sample individually, prior to merging samples for joint analysis)</span>
<span class="va">PBMCs.list</span> <span class="op">&lt;-</span> <span class="fu">SplitObject</span><span class="op">(</span><span class="va">PBMCs</span>, split.by <span class="op">=</span> <span class="st">"orig.ident"</span><span class="op">)</span>
<span class="va">features.list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="op">)</span>
<span class="co"># Normalize data &amp; select highly variable transcripts</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">PBMCs.list</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">PBMCs.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">PBMCs.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span> 
                       <span class="fu">NormalizeData</span><span class="op">(</span>normalization.method <span class="op">=</span> <span class="st">"LogNormalize"</span>, 
                                     assay <span class="op">=</span> <span class="st">"RNA"</span>, scale.factor <span class="op">=</span> <span class="fl">10000</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span> 
                       <span class="fu">NormalizeData</span><span class="op">(</span>normalization.method <span class="op">=</span> <span class="st">"CLR"</span>, 
                                     assay <span class="op">=</span> <span class="st">"ADT"</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span>
                       <span class="fu">FindVariableFeatures</span><span class="op">(</span>selection.method <span class="op">=</span> <span class="st">"vst"</span>, 
                                            nfeatures <span class="op">=</span> <span class="fl">2000</span><span class="op">)</span>
    <span class="va">features.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">PBMCs.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">@</span><span class="va">var.features</span>
<span class="op">}</span>
<span class="co"># Merge into a cohesive list of variable transcripts, keeping only those which are shared between all individual lists</span>
<span class="va">var.features</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/funprog.html">Reduce</a></span><span class="op">(</span><span class="va">intersect</span>, <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">features.list</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,
                                   <span class="va">features.list</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span>,
                                   <span class="va">features.list</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
<span class="co"># Merge all three samples into a single object for joint analysis of cells</span>
<span class="va">PBMCs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html">merge</a></span><span class="op">(</span><span class="va">PBMCs.list</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">PBMCs.list</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span>, <span class="va">PBMCs.list</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,
               project <span class="op">=</span> <span class="st">"PBMCs"</span><span class="op">)</span>
<span class="co"># Update variable genes in object</span>
<span class="va">PBMCs</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">@</span><span class="va">var.features</span> <span class="op">&lt;-</span> <span class="va">var.features</span>
<span class="co"># Scale data and generate PBMCs</span>
<span class="va">PBMCs</span> <span class="op">&lt;-</span> <span class="va">PBMCs</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span>
         <span class="fu">ScaleData</span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span> 
         <span class="fu">RunPCA</span><span class="op">(</span>npcs <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></code></pre></div>
</div>
<div id="selecting-the-appropriate-number-of-principal-components-for-umap-reduction" class="section level3">
<h3 class="hasAnchor">
<a href="#selecting-the-appropriate-number-of-principal-components-for-umap-reduction" class="anchor"></a>Selecting the appropriate number of Principal Components for UMAP reduction</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Identify correct numbers of PCs</span>
<span class="co">## (Takes up to 5 minutes. Not run while rendering vignette for time)</span>

<span class="co"># PBMCs.pca.test &lt;- TestPCA(PBMCs)</span>
<span class="co"># PBMCs.pca.test[, 1:20]</span>

<span class="co">## 9 PCs with z &gt; 1</span>
<span class="co">## Proceed with 9 PCs for dimensional reduction &amp; clustering</span>
<span class="co">## Visualize PCs plotted by standard deviation:</span>
<span class="fu">ElbowPlot</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span></code></pre></div>
<p><img src="NormalizationMethods_files/figure-html/PC_Selection_logscale_indep-1.png" width="672" style="display: block; margin: auto;"></p>
</div>
<div id="generating-a-umap-and-clusters" class="section level3">
<h3 class="hasAnchor">
<a href="#generating-a-umap-and-clusters" class="anchor"></a>Generating a UMAP and clusters</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">PBMCs</span> <span class="op">&lt;-</span> <span class="va">PBMCs</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span> 
         <span class="fu">RunUMAP</span><span class="op">(</span>reduction <span class="op">=</span> <span class="st">"pca"</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">9</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span>
         <span class="fu">FindNeighbors</span><span class="op">(</span>reduction <span class="op">=</span> <span class="st">"pca"</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">9</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span>
         <span class="fu">FindClusters</span><span class="op">(</span>resolution <span class="op">=</span> <span class="fl">.8</span><span class="op">)</span></code></pre></div>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 16946
## Number of edges: 551155
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.8846
## Number of communities: 14
## Elapsed time: 2 seconds</code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">UMAPPlot</span><span class="op">(</span><span class="va">PBMCs</span>, cols <span class="op">=</span> <span class="va">colors.use</span>, group.by <span class="op">=</span> <span class="st">"orig.ident"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Log-Normalize + Scale (Prior to merging) \nSample ID"</span><span class="op">)</span></code></pre></div>
<p><img src="NormalizationMethods_files/figure-html/Clustering_logscale_indep-1.png" width="672" style="display: block; margin: auto;"></p>
</div>
<div id="measuring-sample-umap-integration-generating-an-ilisi-score" class="section level3">
<h3 class="hasAnchor">
<a href="#measuring-sample-umap-integration-generating-an-ilisi-score" class="anchor"></a>Measuring sample-UMAP integration (generating an iLISI score)</h3>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/GetiLISI.html">GetiLISI</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMCs</span>, nSamples <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0.210613</code></pre>
</div>
</div>
<div id="cell-typing-of-joint-pbmcs-object-for-cms" class="section level2">
<h2 class="hasAnchor">
<a href="#cell-typing-of-joint-pbmcs-object-for-cms" class="anchor"></a>Cell Typing of joint PBMCs object for CMS</h2>
<div id="convert-cluster-classifications-to-cell-type-classifications" class="section level3">
<h3 class="hasAnchor">
<a href="#convert-cluster-classifications-to-cell-type-classifications" class="anchor"></a>Convert cluster classifications to cell type classifications</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># For complete cell classification workflow see our vignette "Biaxial Gating of a Single Sample"</span>
<span class="co"># More details can be found in figure S3 of our manuscript "Data Matrix Normalization and Merging Strategies Minimize Batch-specific Systemic Variation in scRNA-Seq Data."</span>
<span class="fu">UMAPPlot</span><span class="op">(</span><span class="va">PBMCs</span>, cols <span class="op">=</span> <span class="va">colors.use</span>, pt.size <span class="op">=</span> <span class="fl">2</span>, 
         group.by <span class="op">=</span> <span class="st">"seurat_clusters"</span>, label <span class="op">=</span> <span class="cn">T</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Log-Normalize + Scale (Prior to merging) \nClusters"</span><span class="op">)</span></code></pre></div>
<p><img src="NormalizationMethods_files/figure-html/Cluster_assignment_logscale_indep-1.png" width="672" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># B_Cells = 8</span>
<span class="co"># T_CD4 = 0, 1, 13</span>
<span class="co"># No TReg: Within cluster 1</span>
<span class="co"># T_CD8 = 2, 7</span>
<span class="co"># NK_T = 5</span>
<span class="co"># NK = 3</span>
<span class="co"># NK_CD56Hi = 10</span>
<span class="co"># Monocyte_Classical = 4, 6</span>
<span class="co"># Monocyte_NonClassical = 9</span>
<span class="co"># Dendritic_Cells = 11, 12</span>
<span class="co"># HSPCs = no cluster</span>
<span class="co"># Cycling_Cells = no cluster</span>
<span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">PBMCs</span><span class="op">[[</span><span class="st">"seurat_clusters"</span><span class="op">]</span><span class="op">]</span>
<span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">plyr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyr/man/mapvalues.html">mapvalues</a></span><span class="op">(</span><span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span>, from <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">8</span>, <span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">13</span>,
                                                         <span class="fl">2</span>, <span class="fl">7</span>, <span class="fl">5</span>, <span class="fl">3</span>, <span class="fl">10</span>,
                                                         <span class="fl">4</span>, <span class="fl">6</span>, 
                                                         <span class="fl">9</span>, <span class="fl">11</span>, <span class="fl">12</span><span class="op">)</span>, 
                to <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'B_Cells'</span>, <span class="st">'T_CD4'</span>, <span class="st">'T_CD4'</span>, <span class="st">'T_CD4'</span>,
                       <span class="st">'T_CD8'</span>, <span class="st">'T_CD8'</span>, <span class="st">'NK_T'</span>, <span class="st">"NK"</span>, <span class="st">"NK_CD56Hi"</span>,
                       <span class="st">'Monocyte_Classical'</span>, <span class="st">'Monocyte_Classical'</span>, 
                       <span class="st">'Monocyte_NonClassical'</span>, <span class="st">'Dendritic_Cells'</span>, <span class="st">'Dendritic_Cells'</span><span class="op">)</span><span class="op">)</span>
<span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span>,
                                 levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"B_Cells"</span>, <span class="st">"T_CD4"</span>, <span class="st">"TReg"</span>, 
                                            <span class="st">"T_CD8"</span>, <span class="st">"NK_T"</span>, <span class="st">"NK"</span>, <span class="st">"NK_CD56Hi"</span>,
                                            <span class="st">"Monocyte_Classical"</span>, <span class="st">"Monocyte_NonClassical"</span>,
                                            <span class="st">"Dendritic_Cells"</span>, <span class="st">"HSPCs"</span>, <span class="st">"Cycling_Cells"</span><span class="op">)</span><span class="op">)</span>
<span class="va">PBMCs</span><span class="op">[[</span><span class="st">"Cell_Type"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span>
<span class="fu">UMAPPlot</span><span class="op">(</span><span class="va">PBMCs</span>, cols <span class="op">=</span> <span class="va">colors.use</span>, label <span class="op">=</span> <span class="cn">F</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Log-Normalize + Scale (Prior to merging) \nCell type classifications"</span><span class="op">)</span></code></pre></div>
<p><img src="NormalizationMethods_files/figure-html/Cluster_assignment_logscale_indep-2.png" width="672" style="display: block; margin: auto;"></p>
</div>
</div>
<div id="compare-single-sample-workflow-cell-type-classifications-to-joint-classifications-generating-a-cms" class="section level2">
<h2 class="hasAnchor">
<a href="#compare-single-sample-workflow-cell-type-classifications-to-joint-classifications-generating-a-cms" class="anchor"></a>Compare single-sample workflow cell type classifications to joint classifications (generating a CMS)</h2>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># PBMC Sample 1</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">PBMC1_Single_ID</span><span class="op">)</span>
<span class="va">S1cms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetCMS.html">GetCMS</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMCs</span>, 
                sample.ID <span class="op">=</span> <span class="st">"Sample_01"</span>, 
                reference.ID <span class="op">=</span> <span class="va">PBMC1_Single_ID</span><span class="op">)</span>
<span class="co"># PBMC Sample 2</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">PBMC2_Single_ID</span><span class="op">)</span>
<span class="va">S2cms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetCMS.html">GetCMS</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMCs</span>, 
                sample.ID <span class="op">=</span> <span class="st">"Sample_02"</span>, 
                reference.ID <span class="op">=</span> <span class="va">PBMC2_Single_ID</span><span class="op">)</span>
<span class="co"># PBMC Sample 3</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">PBMC3_Single_ID</span><span class="op">)</span>
<span class="va">S3cms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetCMS.html">GetCMS</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMCs</span>,
                sample.ID <span class="op">=</span> <span class="st">"Sample_03"</span>,
                reference.ID <span class="op">=</span> <span class="va">PBMC3_Single_ID</span><span class="op">)</span>

<span class="co"># Average CMS</span>
<span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">S1cms</span>, <span class="va">S2cms</span>, <span class="va">S3cms</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0.1598935</code></pre>
</div>
</div>
<div id="log-normalize-scaling-figure-5a-1" class="section level1">
<h1 class="hasAnchor">
<a href="#log-normalize-scaling-figure-5a-1" class="anchor"></a>Log-Normalize + Scaling (Figure 5A)</h1>
<div id="after-merging-joined-samples" class="section level2">
<h2 class="hasAnchor">
<a href="#after-merging-joined-samples" class="anchor"></a>After merging (joined samples)</h2>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># This is an identical workflow to that presented in the vignette titled "Sample Donor Effects (Figure 2A)."</span>
<span class="co"># For brevity, I have not duplicated this workflow again here.</span>
<span class="co"># To see details on that analysis (the results of which are also summarised in figure 5A, right panel) please view that vignette and our manuscript.</span></code></pre></div>
</div>
</div>
<div id="sctransform-normalization-figure-5b" class="section level1">
<h1 class="hasAnchor">
<a href="#sctransform-normalization-figure-5b" class="anchor"></a>SCTransform normalization (Figure 5B)</h1>
<div id="prior-to-merging-individual-samples-1" class="section level2">
<h2 class="hasAnchor">
<a href="#prior-to-merging-individual-samples-1" class="anchor"></a>Prior to merging (individual samples)</h2>
<div id="data-filtering-normalization-1" class="section level3">
<h3 class="hasAnchor">
<a href="#data-filtering-normalization-1" class="anchor"></a>Data filtering &amp; normalization</h3>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ghosn-lab.github.io/BatchNorm">BatchNorm</a></span><span class="op">)</span>
<span class="co"># Import unfiltered Seurat object (included with 'BatchNorm' package)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span>

<span class="co"># Filter sample by mitochondrial percentage (+5 SD)</span>
<span class="va">PBMCs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MitoFilter.html">MitoFilter</a></span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span>
<span class="co"># Run "standard" Seurat workflow on a list of samples, independently:</span>
<span class="co"># Including data normalization, variable gene selection and gene scaling (performed on each sample individually, prior to merging samples for joint analysis)</span>
<span class="va">PBMCs.list</span> <span class="op">&lt;-</span> <span class="fu">SplitObject</span><span class="op">(</span><span class="va">PBMCs</span>, split.by <span class="op">=</span> <span class="st">"orig.ident"</span><span class="op">)</span>
<span class="va">features.list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="op">)</span>
<span class="co"># Apply SCTransform to normalize data &amp; select highly variable transcripts</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">PBMCs.list</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">PBMCs.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">PBMCs.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span>
                       <span class="fu"><a href="../reference/MitoFilter.html">MitoFilter</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span> 
                       <span class="fu">NormalizeData</span><span class="op">(</span>verbose <span class="op">=</span> <span class="cn">FALSE</span>, assay <span class="op">=</span> <span class="st">"ADT"</span>, 
                                     normalization.method <span class="op">=</span> <span class="st">"CLR"</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span> 
                       <span class="fu">SCTransform</span><span class="op">(</span>verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span>
                       <span class="fu">RunPCA</span><span class="op">(</span>npcs <span class="op">=</span> <span class="fl">30</span><span class="op">)</span>
    <span class="va">features.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">PBMCs.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">SCT</span><span class="op">@</span><span class="va">var.features</span>
<span class="op">}</span>
<span class="co"># Merge into a cohesive list of variable transcripts, keeping only those which are shared between all individual lists</span>
<span class="va">var.features</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/funprog.html">Reduce</a></span><span class="op">(</span><span class="va">intersect</span>, <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">features.list</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,
                                   <span class="va">features.list</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span>,
                                   <span class="va">features.list</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
<span class="co"># Merge all three samples into a single object for joint analysis of cells</span>
<span class="va">PBMCs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html">merge</a></span><span class="op">(</span><span class="va">PBMCs.list</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">PBMCs.list</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span>, <span class="va">PBMCs.list</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,
               project <span class="op">=</span> <span class="st">"PBMCs"</span><span class="op">)</span>
<span class="co"># Update variable genes in object</span>
<span class="va">PBMCs</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">SCT</span><span class="op">@</span><span class="va">var.features</span> <span class="op">&lt;-</span> <span class="va">var.features</span>
<span class="co"># Scale data and generate PBMCs</span>
<span class="va">PBMCs</span> <span class="op">&lt;-</span> <span class="va">PBMCs</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span>
         <span class="fu">RunPCA</span><span class="op">(</span>npcs <span class="op">=</span> <span class="fl">30</span>, assay <span class="op">=</span> <span class="st">"SCT"</span><span class="op">)</span></code></pre></div>
</div>
<div id="selecting-the-appropriate-number-of-principal-components-for-umap-reduction-1" class="section level3">
<h3 class="hasAnchor">
<a href="#selecting-the-appropriate-number-of-principal-components-for-umap-reduction-1" class="anchor"></a>Selecting the appropriate number of Principal Components for UMAP reduction</h3>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Identify correct numbers of PCs</span>
<span class="co">## (Takes up to 5 minutes. Not run while rendering vignette for time)</span>

<span class="co"># PBMCs.pca.test &lt;- TestPCA(genes.use = PBMCs@assays$SCT@var.features,</span>
<span class="co">#                           mtx.use = PBMCs@assays$SCT@scale.data)</span>
<span class="co"># PBMCs.pca.test[, 1:20]</span>

<span class="co">## 9 PCs with z &gt; 1</span>
<span class="co">## Proceed with 9 PCs for dimensional reduction &amp; clustering</span>
<span class="co">## Visualize PCs plotted by standard deviation:</span>
<span class="fu">ElbowPlot</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span></code></pre></div>
<p><img src="NormalizationMethods_files/figure-html/PC_Selection_SCTindep-1.png" width="672" style="display: block; margin: auto;"></p>
</div>
<div id="generating-a-umap-and-clusters-1" class="section level3">
<h3 class="hasAnchor">
<a href="#generating-a-umap-and-clusters-1" class="anchor"></a>Generating a UMAP and clusters</h3>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">PBMCs</span> <span class="op">&lt;-</span> <span class="va">PBMCs</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span> 
         <span class="fu">RunUMAP</span><span class="op">(</span>reduction <span class="op">=</span> <span class="st">"pca"</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">9</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span>
         <span class="fu">FindNeighbors</span><span class="op">(</span>reduction <span class="op">=</span> <span class="st">"pca"</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">9</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span>
         <span class="fu">FindClusters</span><span class="op">(</span>resolution <span class="op">=</span> <span class="fl">.8</span><span class="op">)</span></code></pre></div>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 16887
## Number of edges: 541494
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.8892
## Number of communities: 18
## Elapsed time: 2 seconds</code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">UMAPPlot</span><span class="op">(</span><span class="va">PBMCs</span>, cols <span class="op">=</span> <span class="va">colors.use</span>, group.by <span class="op">=</span> <span class="st">"orig.ident"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"SCTransform (Prior to merging) \nSample ID"</span><span class="op">)</span></code></pre></div>
<p><img src="NormalizationMethods_files/figure-html/Clustering_SCTindep-1.png" width="672" style="display: block; margin: auto;"></p>
</div>
<div id="measuring-sample-umap-integration-generating-an-ilisi-score-1" class="section level3">
<h3 class="hasAnchor">
<a href="#measuring-sample-umap-integration-generating-an-ilisi-score-1" class="anchor"></a>Measuring sample-UMAP integration (generating an iLISI score)</h3>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/GetiLISI.html">GetiLISI</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMCs</span>, nSamples <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0.4156777</code></pre>
</div>
</div>
<div id="cell-typing-of-joint-pbmcs-object-for-cms-1" class="section level2">
<h2 class="hasAnchor">
<a href="#cell-typing-of-joint-pbmcs-object-for-cms-1" class="anchor"></a>Cell Typing of joint PBMCs object for CMS</h2>
<div id="convert-cluster-classifications-to-cell-type-classifications-1" class="section level3">
<h3 class="hasAnchor">
<a href="#convert-cluster-classifications-to-cell-type-classifications-1" class="anchor"></a>Convert cluster classifications to cell type classifications</h3>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># For complete cell classification workflow see our vignette "Biaxial Gating of a Single Sample"</span>
<span class="co"># More details can be found in figure S3 of our manuscript "Data Matrix Normalization and Merging Strategies Minimize Batch-specific Systemic Variation in scRNA-Seq Data."</span>
<span class="fu">UMAPPlot</span><span class="op">(</span><span class="va">PBMCs</span>, cols <span class="op">=</span> <span class="va">colors.use</span>, pt.size <span class="op">=</span> <span class="fl">2</span>, 
         group.by <span class="op">=</span> <span class="st">"seurat_clusters"</span>, label <span class="op">=</span> <span class="cn">T</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"SCTransform (Prior to merging) \nClusters"</span><span class="op">)</span></code></pre></div>
<p><img src="NormalizationMethods_files/figure-html/Cluster_assignment_SCTindep-1.png" width="672" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># B_Cells = 11</span>
<span class="co"># T_CD4 = 0, 6, 8, 10, 15</span>
<span class="co"># No TReg: Within cluster 10</span>
<span class="co"># T_CD8 = 3, 4, 9, 14</span>
<span class="co"># NK_T = 7</span>
<span class="co"># NK = 1</span>
<span class="co"># NK_CD56Hi = 13</span>
<span class="co"># Monocyte_Classical = 2, 5</span>
<span class="co"># Monocyte_NonClassical = 12</span>
<span class="co"># Dendritic_Cells = 16, 17</span>
<span class="co"># HSPCs = no cluster</span>
<span class="co"># Cycling_Cells = no cluster</span>
<span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">PBMCs</span><span class="op">[[</span><span class="st">"seurat_clusters"</span><span class="op">]</span><span class="op">]</span>
<span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">plyr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyr/man/mapvalues.html">mapvalues</a></span><span class="op">(</span><span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span>, from <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">11</span>, <span class="fl">0</span>, <span class="fl">6</span>, <span class="fl">8</span>, <span class="fl">10</span>, <span class="fl">15</span>,
                                                         <span class="fl">3</span>, <span class="fl">4</span>, <span class="fl">9</span>, <span class="fl">14</span>, <span class="fl">7</span>,
                                                         <span class="fl">1</span>, <span class="fl">13</span>, 
                                                         <span class="fl">2</span>, <span class="fl">5</span>,
                                                         <span class="fl">12</span>, <span class="fl">16</span>, <span class="fl">17</span><span class="op">)</span>, 
                to <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'B_Cells'</span>, <span class="st">'T_CD4'</span>, <span class="st">'T_CD4'</span>, <span class="st">'T_CD4'</span>, <span class="st">'T_CD4'</span>, <span class="st">'T_CD4'</span>,
                       <span class="st">'T_CD8'</span>, <span class="st">'T_CD8'</span>, <span class="st">'T_CD8'</span>, <span class="st">'T_CD8'</span>, <span class="st">'NK_T'</span>, 
                       <span class="st">"NK"</span>, <span class="st">"NK_CD56Hi"</span>,
                       <span class="st">'Monocyte_Classical'</span>, <span class="st">'Monocyte_Classical'</span>, 
                       <span class="st">'Monocyte_NonClassical'</span>, <span class="st">'Dendritic_Cells'</span>, <span class="st">'Dendritic_Cells'</span><span class="op">)</span><span class="op">)</span>
<span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span>,
                                 levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"B_Cells"</span>, <span class="st">"T_CD4"</span>, <span class="st">"TReg"</span>, 
                                            <span class="st">"T_CD8"</span>, <span class="st">"NK_T"</span>, <span class="st">"NK"</span>, <span class="st">"NK_CD56Hi"</span>,
                                            <span class="st">"Monocyte_Classical"</span>, <span class="st">"Monocyte_NonClassical"</span>,
                                            <span class="st">"Dendritic_Cells"</span>, <span class="st">"HSPCs"</span>, <span class="st">"Cycling_Cells"</span><span class="op">)</span><span class="op">)</span>
<span class="va">PBMCs</span><span class="op">[[</span><span class="st">"Cell_Type"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span>
<span class="fu">UMAPPlot</span><span class="op">(</span><span class="va">PBMCs</span>, cols <span class="op">=</span> <span class="va">colors.use</span>, label <span class="op">=</span> <span class="cn">F</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"SCTransform (Prior to merging) \nCell type classifications"</span><span class="op">)</span></code></pre></div>
<p><img src="NormalizationMethods_files/figure-html/Cluster_assignment_SCTindep-2.png" width="672" style="display: block; margin: auto;"></p>
</div>
</div>
<div id="compare-single-sample-workflow-cell-type-classifications-to-joint-classifications-generating-a-cms-1" class="section level2">
<h2 class="hasAnchor">
<a href="#compare-single-sample-workflow-cell-type-classifications-to-joint-classifications-generating-a-cms-1" class="anchor"></a>Compare single-sample workflow cell type classifications to joint classifications (generating a CMS)</h2>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># PBMC Sample 1</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">PBMC1_Single_ID</span><span class="op">)</span>
<span class="va">S1cms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetCMS.html">GetCMS</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMCs</span>, 
                sample.ID <span class="op">=</span> <span class="st">"Sample_01"</span>, 
                reference.ID <span class="op">=</span> <span class="va">PBMC1_Single_ID</span><span class="op">)</span>
<span class="co"># PBMC Sample 2</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">PBMC2_Single_ID</span><span class="op">)</span>
<span class="va">S2cms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetCMS.html">GetCMS</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMCs</span>, 
                sample.ID <span class="op">=</span> <span class="st">"Sample_02"</span>, 
                reference.ID <span class="op">=</span> <span class="va">PBMC2_Single_ID</span><span class="op">)</span>
<span class="co"># PBMC Sample 3</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">PBMC3_Single_ID</span><span class="op">)</span>
<span class="va">S3cms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetCMS.html">GetCMS</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMCs</span>,
                sample.ID <span class="op">=</span> <span class="st">"Sample_03"</span>,
                reference.ID <span class="op">=</span> <span class="va">PBMC3_Single_ID</span><span class="op">)</span>

<span class="co"># Average CMS</span>
<span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">S1cms</span>, <span class="va">S2cms</span>, <span class="va">S3cms</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0.2020373</code></pre>
</div>
</div>
<div id="sctransform-normalization-figure-5b-1" class="section level1">
<h1 class="hasAnchor">
<a href="#sctransform-normalization-figure-5b-1" class="anchor"></a>SCTransform normalization (Figure 5B)</h1>
<div id="after-merging-joined-samples-1" class="section level2">
<h2 class="hasAnchor">
<a href="#after-merging-joined-samples-1" class="anchor"></a>After merging (joined samples)</h2>
<div id="data-filtering-normalization-2" class="section level3">
<h3 class="hasAnchor">
<a href="#data-filtering-normalization-2" class="anchor"></a>Data filtering &amp; normalization</h3>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ghosn-lab.github.io/BatchNorm">BatchNorm</a></span><span class="op">)</span>
<span class="co"># Import unfiltered Seurat object (included with 'BatchNorm' package)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span>

<span class="co"># Run "standard" Seurat workflow: "https://satijalab.org/seurat/articles/pbmc3k_tutorial.html"</span>
<span class="co"># Including a filter of sample by mitochondrial percentage (+5 SD)</span>
<span class="va">PBMCs</span> <span class="op">&lt;-</span> <span class="va">PBMCs</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span> 
         <span class="fu"><a href="../reference/MitoFilter.html">MitoFilter</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span> 
         <span class="fu">NormalizeData</span><span class="op">(</span>verbose <span class="op">=</span> <span class="cn">FALSE</span>, assay <span class="op">=</span> <span class="st">"ADT"</span>, normalization.method <span class="op">=</span> <span class="st">"CLR"</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span> 
         <span class="fu">SCTransform</span><span class="op">(</span>verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span>
         <span class="fu">RunPCA</span><span class="op">(</span>npcs <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></code></pre></div>
</div>
<div id="selecting-the-appropriate-number-of-principal-components-for-umap-reduction-2" class="section level3">
<h3 class="hasAnchor">
<a href="#selecting-the-appropriate-number-of-principal-components-for-umap-reduction-2" class="anchor"></a>Selecting the appropriate number of Principal Components for UMAP reduction</h3>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Identify correct numbers of PCs</span>
<span class="co">## (Takes up to 5 minutes. Not run while rendering vignette for time)</span>

<span class="co"># PBMCs.pca.test &lt;- TestPCA(genes.use = PBMCs@assays$SCT@var.features,</span>
<span class="co">#                           mtx.use = PBMCs@assays$SCT@scale.data)</span>
<span class="co"># PBMCs.pca.test[, 1:20]</span>

<span class="co">## 14 PCs with z &gt; 1</span>
<span class="co">## Proceed with 14 PCs for dimensional reduction &amp; clustering</span>
<span class="co">## Visualize PCs plotted by standard deviation:</span>
<span class="fu">ElbowPlot</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span></code></pre></div>
<p><img src="NormalizationMethods_files/figure-html/PC_Selection_SCTmerge-1.png" width="672" style="display: block; margin: auto;"></p>
</div>
<div id="generating-a-umap-and-clusters-2" class="section level3">
<h3 class="hasAnchor">
<a href="#generating-a-umap-and-clusters-2" class="anchor"></a>Generating a UMAP and clusters</h3>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">PBMCs</span> <span class="op">&lt;-</span> <span class="va">PBMCs</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span> 
         <span class="fu">RunUMAP</span><span class="op">(</span>reduction <span class="op">=</span> <span class="st">"pca"</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">14</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span>
         <span class="fu">FindNeighbors</span><span class="op">(</span>reduction <span class="op">=</span> <span class="st">"pca"</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">14</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span>
         <span class="fu">FindClusters</span><span class="op">(</span>resolution <span class="op">=</span> <span class="fl">.8</span><span class="op">)</span></code></pre></div>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 16946
## Number of edges: 574465
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.8858
## Number of communities: 18
## Elapsed time: 2 seconds</code></pre>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">UMAPPlot</span><span class="op">(</span><span class="va">PBMCs</span>, cols <span class="op">=</span> <span class="va">colors.use</span>, group.by <span class="op">=</span> <span class="st">"orig.ident"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"SCTransform (After merging) \nSample ID"</span><span class="op">)</span></code></pre></div>
<p><img src="NormalizationMethods_files/figure-html/Clustering_SCTmerge-1.png" width="672" style="display: block; margin: auto;"></p>
</div>
<div id="measuring-sample-umap-integration-generating-an-ilisi-score-2" class="section level3">
<h3 class="hasAnchor">
<a href="#measuring-sample-umap-integration-generating-an-ilisi-score-2" class="anchor"></a>Measuring sample-UMAP integration (generating an iLISI score)</h3>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/GetiLISI.html">GetiLISI</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMCs</span>, nSamples <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0.5585073</code></pre>
</div>
</div>
<div id="cell-typing-of-joint-pbmcs-object-for-cms-2" class="section level2">
<h2 class="hasAnchor">
<a href="#cell-typing-of-joint-pbmcs-object-for-cms-2" class="anchor"></a>Cell Typing of joint PBMCs object for CMS</h2>
<div id="convert-cluster-classifications-to-cell-type-classifications-2" class="section level3">
<h3 class="hasAnchor">
<a href="#convert-cluster-classifications-to-cell-type-classifications-2" class="anchor"></a>Convert cluster classifications to cell type classifications</h3>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># For complete cell classification workflow see our vignette "Biaxial Gating of a Single Sample"</span>
<span class="co"># More details can be found in figure S3 of our manuscript "Data Matrix Normalization and Merging Strategies Minimize Batch-specific Systemic Variation in scRNA-Seq Data."</span>
<span class="fu">UMAPPlot</span><span class="op">(</span><span class="va">PBMCs</span>, cols <span class="op">=</span> <span class="va">colors.use</span>, pt.size <span class="op">=</span> <span class="fl">2</span>, 
         group.by <span class="op">=</span> <span class="st">"seurat_clusters"</span>, label <span class="op">=</span> <span class="cn">T</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"SCTransform (After merging) \nClusters"</span><span class="op">)</span></code></pre></div>
<p><img src="NormalizationMethods_files/figure-html/Cluster_assignment_SCTmerge-1.png" width="672" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># B_Cells = 10</span>
<span class="co"># T_CD4 = 0, 4</span>
<span class="co"># TReg = 11</span>
<span class="co"># T_CD8 = 6, 7, 8, 9</span>
<span class="co"># NK_T = 2</span>
<span class="co"># NK = 1</span>
<span class="co"># NKCD56Hi = 14</span>
<span class="co"># Monocyte_Classical = 3, 5, 13</span>
<span class="co"># Monocyte_NonClassical = 12</span>
<span class="co"># Dendritic_Cells = 16, 17</span>
<span class="co"># HSPCs = 15</span>
<span class="co"># No Cycling_Cells. Within NK cluster</span>
<span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">PBMCs</span><span class="op">[[</span><span class="st">"seurat_clusters"</span><span class="op">]</span><span class="op">]</span>
<span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">plyr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyr/man/mapvalues.html">mapvalues</a></span><span class="op">(</span><span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span>, from <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">0</span>, <span class="fl">4</span>, <span class="fl">11</span>, <span class="fl">6</span>,
                                                         <span class="fl">7</span>, <span class="fl">8</span>, <span class="fl">9</span>, <span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">14</span>,
                                                         <span class="fl">3</span>, <span class="fl">5</span>, <span class="fl">13</span>, 
                                                         <span class="fl">12</span>, <span class="fl">16</span>, <span class="fl">17</span>,
                                                         <span class="fl">15</span><span class="op">)</span>, 
                to <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'B_Cells'</span>, <span class="st">'T_CD4'</span>, <span class="st">'T_CD4'</span>, <span class="st">'TReg'</span>, <span class="st">'T_CD8'</span>, 
                       <span class="st">'T_CD8'</span>, <span class="st">'T_CD8'</span>, <span class="st">'T_CD8'</span>, <span class="st">'NK_T'</span>, <span class="st">"NK"</span>, <span class="st">'NK_CD56Hi'</span>,
                       <span class="st">'Monocyte_Classical'</span>, <span class="st">'Monocyte_Classical'</span>, <span class="st">'Monocyte_Classical'</span>,
                       <span class="st">'Monocyte_NonClassical'</span>, <span class="st">'Dendritic_Cells'</span>, <span class="st">'Dendritic_Cells'</span>,
                       <span class="st">'HSPCs'</span><span class="op">)</span><span class="op">)</span>
<span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span>,
                                 levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"B_Cells"</span>, <span class="st">"T_CD4"</span>, <span class="st">"TReg"</span>, 
                                            <span class="st">"T_CD8"</span>, <span class="st">"NK_T"</span>, <span class="st">"NK"</span>, <span class="st">"NK_CD56Hi"</span>,
                                            <span class="st">"Monocyte_Classical"</span>, <span class="st">"Monocyte_NonClassical"</span>,
                                            <span class="st">"Dendritic_Cells"</span>, <span class="st">"HSPCs"</span>, <span class="st">"Cycling_Cells"</span><span class="op">)</span><span class="op">)</span>
<span class="va">PBMCs</span><span class="op">[[</span><span class="st">"Cell_Type"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span>
<span class="fu">UMAPPlot</span><span class="op">(</span><span class="va">PBMCs</span>, cols <span class="op">=</span> <span class="va">colors.use</span>, label <span class="op">=</span> <span class="cn">F</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"SCTransform (After merging) \nCell type classifications"</span><span class="op">)</span></code></pre></div>
<p><img src="NormalizationMethods_files/figure-html/Cluster_assignment_SCTmerge-2.png" width="672" style="display: block; margin: auto;"></p>
</div>
</div>
<div id="compare-single-sample-workflow-cell-type-classifications-to-joint-classifications-generating-a-cms-2" class="section level2">
<h2 class="hasAnchor">
<a href="#compare-single-sample-workflow-cell-type-classifications-to-joint-classifications-generating-a-cms-2" class="anchor"></a>Compare single-sample workflow cell type classifications to joint classifications (generating a CMS)</h2>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># PBMC Sample 1</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">PBMC1_Single_ID</span><span class="op">)</span>
<span class="va">S1cms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetCMS.html">GetCMS</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMCs</span>, 
                sample.ID <span class="op">=</span> <span class="st">"Sample_01"</span>, 
                reference.ID <span class="op">=</span> <span class="va">PBMC1_Single_ID</span><span class="op">)</span>
<span class="co"># PBMC Sample 2</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">PBMC2_Single_ID</span><span class="op">)</span>
<span class="va">S2cms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetCMS.html">GetCMS</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMCs</span>, 
                sample.ID <span class="op">=</span> <span class="st">"Sample_02"</span>, 
                reference.ID <span class="op">=</span> <span class="va">PBMC2_Single_ID</span><span class="op">)</span>
<span class="co"># PBMC Sample 3</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">PBMC3_Single_ID</span><span class="op">)</span>
<span class="va">S3cms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetCMS.html">GetCMS</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMCs</span>,
                sample.ID <span class="op">=</span> <span class="st">"Sample_03"</span>,
                reference.ID <span class="op">=</span> <span class="va">PBMC3_Single_ID</span><span class="op">)</span>

<span class="co"># Average CMS</span>
<span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">S1cms</span>, <span class="va">S2cms</span>, <span class="va">S3cms</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0.2040421</code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Benjamin Babcock.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
