<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Batch effects are generated by sample donor â€¢ BatchNorm</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Batch effects are generated by sample donor">
<meta property="og:description" content="How to determine the batch effects introduced by sample donor? scRNA-Seq libraries from independently processed samples may be combined to produce a single analysis. This vignette describes how we determine the batch effects of three identically prepared and sequenced PBMC datasets, using CMS and iLISI scoring. The results of this workflow are depicted in figure 2A.
">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BatchNorm</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BatchEffect_Correction.html">Common batch-effect correction methods imperfectly resolve batch effect</a>
    </li>
    <li>
      <a href="../articles/Biaxial_Gating.html">Biaxial Gating of a Single Sample (Fig. S3)</a>
    </li>
    <li>
      <a href="../articles/NormalizationMethods.html">Data normalization and merging strategies differentially impact the measured batch effect</a>
    </li>
    <li>
      <a href="../articles/Pooled_Sequencing.html">Pooling samples for sequencing does not appreciably improve the measured batch effect</a>
    </li>
    <li>
      <a href="../articles/SampleDonor_Effect.html">Batch effects are generated by sample donor</a>
    </li>
    <li>
      <a href="../articles/Sequencing_Effect.html">Batch effects are generated by sequencing depth, but not sequencing replicates alone</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="SampleDonor_Effect_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Batch effects are generated by sample donor</h1>
                        <h4 class="author">Benjamin R Babcock</h4>
            
      
      
      <div class="hidden name"><code>SampleDonor_Effect.Rmd</code></div>

    </div>

    
    
<div id="processing-the-seurat-object" class="section level1">
<h1 class="hasAnchor">
<a href="#processing-the-seurat-object" class="anchor"></a>Processing the Seurat Object</h1>
<div id="data-filtering-normalization" class="section level2">
<h2 class="hasAnchor">
<a href="#data-filtering-normalization" class="anchor"></a>Data filtering &amp; normalization</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ghosn-lab.github.io/BatchNorm">BatchNorm</a></span><span class="op">)</span>
<span class="co"># Import unfiltered Seurat object (included with 'BatchNorm' package)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span>

<span class="co"># Run "standard" Seurat workflow: </span>
<span class="co"># Including filtering by mitochondrial percentage (+5 SD)</span>
<span class="co"># Including data normalization, variable gene selection and gene scaling (performed on all samples together)</span>
<span class="va">PBMCs</span> <span class="op">&lt;-</span> <span class="va">PBMCs</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span> 
         <span class="fu"><a href="../reference/MitoFilter.html">MitoFilter</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span> 
         <span class="fu">NormalizeData</span><span class="op">(</span>normalization.method <span class="op">=</span> <span class="st">"LogNormalize"</span>, assay <span class="op">=</span> <span class="st">"RNA"</span>, scale.factor <span class="op">=</span> <span class="fl">10000</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span> 
         <span class="fu">NormalizeData</span><span class="op">(</span>verbose <span class="op">=</span> <span class="cn">FALSE</span>, assay <span class="op">=</span> <span class="st">"ADT"</span>, normalization.method <span class="op">=</span> <span class="st">"CLR"</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span>
         <span class="fu">FindVariableFeatures</span><span class="op">(</span>selection.method <span class="op">=</span> <span class="st">"vst"</span>, nfeatures <span class="op">=</span> <span class="fl">2000</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span>
         <span class="fu">ScaleData</span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span> 
         <span class="fu">RunPCA</span><span class="op">(</span>npcs <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></code></pre></div>
</div>
<div id="selecting-the-appropriate-number-of-principal-components-for-umap-reduction" class="section level2">
<h2 class="hasAnchor">
<a href="#selecting-the-appropriate-number-of-principal-components-for-umap-reduction" class="anchor"></a>Selecting the appropriate number of Principal Components for UMAP reduction</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Identify correct numbers of PCs</span>
<span class="co">## (Takes up to 5 minutes. Not run while rendering vignette for time)</span>

<span class="co"># PBMCs.pca.test &lt;- TestPCA(PBMCs)</span>
<span class="co"># PBMCs.pca.test[, 1:20]</span>

<span class="co">## 14 PCs with z &gt; 1</span>
<span class="co">## Proceed with 14 PCs for dimensional reduction &amp; clustering</span>
<span class="co">## Visualize PCs plotted by standard deviation:</span>
<span class="fu">ElbowPlot</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span></code></pre></div>
<p><img src="SampleDonor_Effect_files/figure-html/PC_Selection-1.png" width="672" style="display: block; margin: auto;"></p>
</div>
<div id="generating-a-umap-and-clusters" class="section level2">
<h2 class="hasAnchor">
<a href="#generating-a-umap-and-clusters" class="anchor"></a>Generating a UMAP and clusters</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">PBMCs</span> <span class="op">&lt;-</span> <span class="va">PBMCs</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span> 
         <span class="fu">RunUMAP</span><span class="op">(</span>reduction <span class="op">=</span> <span class="st">"pca"</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">14</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span>
         <span class="fu">FindNeighbors</span><span class="op">(</span>reduction <span class="op">=</span> <span class="st">"pca"</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">14</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html">%&gt;%</a></span>
         <span class="fu">FindClusters</span><span class="op">(</span>resolution <span class="op">=</span> <span class="fl">.8</span><span class="op">)</span></code></pre></div>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 16946
## Number of edges: 593488
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.8927
## Number of communities: 23
## Elapsed time: 2 seconds</code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">UMAPPlot</span><span class="op">(</span><span class="va">PBMCs</span>, cols <span class="op">=</span> <span class="va">colors.use</span>, group.by <span class="op">=</span> <span class="st">"orig.ident"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Default Workflow"</span><span class="op">)</span></code></pre></div>
<p><img src="SampleDonor_Effect_files/figure-html/Clustering-1.png" width="672" style="display: block; margin: auto;"></p>
</div>
<div id="measuring-sample-umap-integration-generating-an-ilisi-score" class="section level2">
<h2 class="hasAnchor">
<a href="#measuring-sample-umap-integration-generating-an-ilisi-score" class="anchor"></a>Measuring sample-UMAP integration (generating an iLISI score)</h2>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/GetiLISI.html">GetiLISI</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMCs</span>, nSamples <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0.8612189</code></pre>
</div>
</div>
<div id="cell-typing-of-joint-pbmcs-object-for-cms" class="section level1">
<h1 class="hasAnchor">
<a href="#cell-typing-of-joint-pbmcs-object-for-cms" class="anchor"></a>Cell Typing of joint PBMCs object for CMS</h1>
<div id="convert-cluster-classifications-to-cell-type-classifications" class="section level2">
<h2 class="hasAnchor">
<a href="#convert-cluster-classifications-to-cell-type-classifications" class="anchor"></a>Convert cluster classifications to cell type classifications</h2>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># For complete cell classification workflow see our vignette "Biaxial Gating of a Single Sample"</span>
<span class="co"># More details can be found in figure S3 of our manuscript "Data Matrix Normalization and Merging Strategies Minimize Batch-specific Systemic Variation in scRNA-Seq Data."</span>
<span class="fu">UMAPPlot</span><span class="op">(</span><span class="va">PBMCs</span>, cols <span class="op">=</span> <span class="va">colors.use</span>, pt.size <span class="op">=</span> <span class="fl">2</span>, 
         group.by <span class="op">=</span> <span class="st">"seurat_clusters"</span>, label <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></code></pre></div>
<p><img src="SampleDonor_Effect_files/figure-html/Cluster_assignment-1.png" width="672" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># B_Cells = 14, 16</span>
<span class="co"># T_CD4 = 0, 1, 3, 21, 19</span>
<span class="co"># No TReg cluster (contained within clusters 1 &amp; 3)</span>
<span class="co"># T_CD8 = 4, 7, 11</span>
<span class="co"># NK_T = 5, 8</span>
<span class="co"># NK = 6, 10</span>
<span class="co"># NKCD56Hi = 15</span>
<span class="co"># Monocyte_Classical = 2, 9, 12, 17</span>
<span class="co"># Monocyte_NonClassical = 13</span>
<span class="co"># Dendritic_Cells = 18</span>
<span class="co"># HSPCs = 20</span>
<span class="co"># Cycling_Cells = 22</span>
<span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">PBMCs</span><span class="op">[[</span><span class="st">"seurat_clusters"</span><span class="op">]</span><span class="op">]</span>
<span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">plyr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyr/man/mapvalues.html">mapvalues</a></span><span class="op">(</span><span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span>, from <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">14</span>, <span class="fl">16</span>, <span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">3</span>, <span class="fl">21</span>, <span class="fl">19</span>,
                                                         <span class="fl">4</span>, <span class="fl">7</span>, <span class="fl">11</span>, <span class="fl">5</span>, <span class="fl">8</span>, <span class="fl">6</span>, <span class="fl">10</span>,
                                                         <span class="fl">15</span>, <span class="fl">2</span>, <span class="fl">9</span>, <span class="fl">12</span>, 
                                                         <span class="fl">17</span>, <span class="fl">13</span>, <span class="fl">18</span>,
                                                         <span class="fl">20</span>, <span class="fl">22</span><span class="op">)</span>, 
                to <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'B_Cells'</span>, <span class="st">'B_Cells'</span>, <span class="st">'T_CD4'</span>, <span class="st">'T_CD4'</span>, <span class="st">'T_CD4'</span>, <span class="st">'T_CD4'</span>, <span class="st">"T_CD4"</span>,
                       <span class="st">'T_CD8'</span>, <span class="st">'T_CD8'</span>, <span class="st">'T_CD8'</span>, <span class="st">'NK_T'</span>, <span class="st">'NK_T'</span>, <span class="st">"NK"</span>, <span class="st">"NK"</span>,
                       <span class="st">'NK_CD56Hi'</span>, <span class="st">'Monocyte_Classical'</span>, <span class="st">'Monocyte_Classical'</span>, <span class="st">'Monocyte_Classical'</span>,
                       <span class="st">'Monocyte_Classical'</span>, <span class="st">'Monocyte_NonClassical'</span>, <span class="st">'Dendritic_Cells'</span>,
                       <span class="st">'HSPCs'</span>, <span class="st">'Cycling_Cells'</span><span class="op">)</span><span class="op">)</span>
<span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span>,
                                 levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"B_Cells"</span>, <span class="st">"T_CD4"</span>, <span class="st">"TReg"</span>, 
                                            <span class="st">"T_CD8"</span>, <span class="st">"NK_T"</span>, <span class="st">"NK"</span>, <span class="st">"NK_CD56Hi"</span>,
                                            <span class="st">"Monocyte_Classical"</span>, <span class="st">"Monocyte_NonClassical"</span>,
                                            <span class="st">"Dendritic_Cells"</span>, <span class="st">"HSPCs"</span>, <span class="st">"Cycling_Cells"</span><span class="op">)</span><span class="op">)</span>
<span class="va">PBMCs</span><span class="op">[[</span><span class="st">"Cell_Type"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">Idents</span><span class="op">(</span><span class="va">PBMCs</span><span class="op">)</span>
<span class="fu">UMAPPlot</span><span class="op">(</span><span class="va">PBMCs</span>, cols <span class="op">=</span> <span class="va">colors.use</span>, label <span class="op">=</span> <span class="cn">F</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Cell Type Classifications"</span><span class="op">)</span></code></pre></div>
<p><img src="SampleDonor_Effect_files/figure-html/Cluster_assignment-2.png" width="672" style="display: block; margin: auto;"></p>
</div>
<div id="compare-single-sample-workflow-cell-type-classifications-to-joint-classifications-generating-a-cms" class="section level2">
<h2 class="hasAnchor">
<a href="#compare-single-sample-workflow-cell-type-classifications-to-joint-classifications-generating-a-cms" class="anchor"></a>Compare single-sample workflow cell type classifications to joint classifications (generating a CMS)</h2>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># PBMC Sample 1</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">PBMC1_Single_ID</span><span class="op">)</span>
<span class="va">S1cms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetCMS.html">GetCMS</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMCs</span>, 
                sample.ID <span class="op">=</span> <span class="st">"Sample_01"</span>, 
                reference.ID <span class="op">=</span> <span class="va">PBMC1_Single_ID</span><span class="op">)</span>
<span class="co"># PBMC Sample 2</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">PBMC2_Single_ID</span><span class="op">)</span>
<span class="va">S2cms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetCMS.html">GetCMS</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMCs</span>, 
                sample.ID <span class="op">=</span> <span class="st">"Sample_02"</span>, 
                reference.ID <span class="op">=</span> <span class="va">PBMC2_Single_ID</span><span class="op">)</span>
<span class="co"># PBMC Sample 3</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">PBMC3_Single_ID</span><span class="op">)</span>
<span class="va">S3cms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetCMS.html">GetCMS</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMCs</span>,
                sample.ID <span class="op">=</span> <span class="st">"Sample_03"</span>,
                reference.ID <span class="op">=</span> <span class="va">PBMC3_Single_ID</span><span class="op">)</span>

<span class="co"># Average CMS</span>
<span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">S1cms</span>, <span class="va">S2cms</span>, <span class="va">S3cms</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0.09812015</code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Benjamin Babcock.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
