<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Sequencing Effects (Figures 2B-D) â€¢ BatchNorm</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Sequencing Effects (Figures 2B-D)">
<meta property="og:description" content="How to determine the batch effects introduced by sequencing replicate and depth? scRNA-Seq libraries generated from two independently processed PBMC samples were sequenced in duplicate (for a total of four sequencing outputs from two PBMC samples). PBMC sample 4 was sequenced twice without altering any parameters, while PBMC sample 5 was sequenced twice to a different depth of sequence. This vignette describes how we use CMS and iLISI scoring to determine the batch effects occuring from identical sequencing at different times, and from sequencing depth. The sample duplicating scheme is depicted in figure 2B, while results of this workflow are depicted in figures 2C-D.
">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BatchNorm</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Biaxial_Gating.html">Biaxial Gating of a Single Sample (Fig. S3)</a>
    </li>
    <li>
      <a href="../articles/SampleDonor_Effect.html">Sample Donor Effects (Figure 2A)</a>
    </li>
    <li>
      <a href="../articles/Sequencing_Effect.html">Sequencing Effects (Figures 2B-D)</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Sequencing_Effect_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Sequencing Effects (Figures 2B-D)</h1>
                        <h4 class="author">Benjamin R Babcock</h4>
            
      
      
      <div class="hidden name"><code>Sequencing_Effect.Rmd</code></div>

    </div>

    
    
<div id="identical-duplicate-sequencing" class="section level1">
<h1 class="hasAnchor">
<a href="#identical-duplicate-sequencing" class="anchor"></a>Identical (duplicate) Sequencing</h1>
<div id="processing-the-seurat-object" class="section level2">
<h2 class="hasAnchor">
<a href="#processing-the-seurat-object" class="anchor"></a>Processing the Seurat Object</h2>
<div id="data-filtering-normalization" class="section level3">
<h3 class="hasAnchor">
<a href="#data-filtering-normalization" class="anchor"></a>Data filtering &amp; normalization</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ghosn-lab.github.io/BatchNorm">BatchNorm</a></span><span class="op">)</span>

<span class="co"># Import unfiltered Seurat object (included with 'BatchNorm' package)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">'BatchNorm'</span>, <span class="va">PBMC4</span><span class="op">)</span>

<span class="co"># Run "standard" Seurat workflow: </span>
<span class="co"># Including filtering by mitochondrial percentage (+5 SD)</span>
<span class="co"># Including data normalization, variable gene selection and gene scaling (performed on all samples together)</span>
<span class="va">PBMC4</span> <span class="op">&lt;-</span> <span class="va">PBMC4</span> <span class="op">%&gt;%</span> 
         <span class="fu"><a href="../reference/MitoFilter.html">MitoFilter</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> 
         <span class="fu">NormalizeData</span><span class="op">(</span>normalization.method <span class="op">=</span> <span class="st">"LogNormalize"</span>, assay <span class="op">=</span> <span class="st">"RNA"</span>, scale.factor <span class="op">=</span> <span class="fl">10000</span><span class="op">)</span> <span class="op">%&gt;%</span> 
         <span class="fu">NormalizeData</span><span class="op">(</span>verbose <span class="op">=</span> <span class="cn">FALSE</span>, assay <span class="op">=</span> <span class="st">"ADT"</span>, normalization.method <span class="op">=</span> <span class="st">"CLR"</span><span class="op">)</span> <span class="op">%&gt;%</span>
         <span class="fu">FindVariableFeatures</span><span class="op">(</span>selection.method <span class="op">=</span> <span class="st">"vst"</span>, nfeatures <span class="op">=</span> <span class="fl">2000</span><span class="op">)</span> <span class="op">%&gt;%</span>
         <span class="fu">ScaleData</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> 
         <span class="fu">RunPCA</span><span class="op">(</span>npcs <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></code></pre></div>
</div>
<div id="selecting-the-appropriate-number-of-principal-components-for-umap-reduction" class="section level3">
<h3 class="hasAnchor">
<a href="#selecting-the-appropriate-number-of-principal-components-for-umap-reduction" class="anchor"></a>Selecting the appropriate number of Principal Components for UMAP reduction</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Identify correct numbers of PCs</span>
<span class="va">PBMC4.pca.test</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/TestPCA.html">TestPCA</a></span><span class="op">(</span><span class="va">PBMC4</span><span class="op">)</span>
<span class="va">PBMC4.pca.test</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">20</span><span class="op">]</span></code></pre></div>
<pre><code>##                   [,1]        [,2]        [,3]        [,4]        [,5]
## SD          8.86559315  7.54140784  6.59695531  5.21211704 3.327591056
## Proportion  0.06400454  0.04631269  0.03543907  0.02212195 0.009016855
## Cumulative  0.06400454  0.11031723  0.14575630  0.16787825 0.176895104
## ZScore     30.92977162 22.31298983 17.01700917 10.53093419 4.148118824
##                   [,6]        [,7]        [,8]        [,9]       [,10]
## SD         3.163380180 2.859949562 2.584769979 2.470122244 2.459866405
## Proportion 0.008148881 0.006660578 0.005440502 0.004968577 0.004927404
## Cumulative 0.185043985 0.191704563 0.197145065 0.202113642 0.207041047
## ZScore     3.725373582 3.000498508 2.406262662 2.176412664 2.156359438
##                 [,11]       [,12]       [,13]       [,14]       [,15]
## SD         2.28032818 2.132632056 2.020492644 1.970615169 1.908314859
## Proportion 0.00423438 0.003703625 0.003324373 0.003162269 0.002965482
## Cumulative 0.21127543 0.214979053 0.218303425 0.221465694 0.224431176
## ZScore     1.81882333 1.560319956 1.375605718 1.296653371 1.200808411
##                  [,16]       [,17]       [,18]       [,19]       [,20]
## SD         1.814874348 1.679251903 1.627326360 1.613357328 1.590671998
## Proportion 0.002682182 0.002296291 0.002156475 0.002119612 0.002060423
## Cumulative 0.227113358 0.229409649 0.231566124 0.233685736 0.235746159
## ZScore     1.062827999 0.874880182 0.806783338 0.788828973 0.760001358</code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># 16 PCs with z &gt; 1</span>
<span class="co"># Proceed with 16 PCs for dimensional reduction &amp; clustering</span>
<span class="co"># Visualize PCs plotted by standard deviation:</span>
<span class="fu">ElbowPlot</span><span class="op">(</span><span class="va">PBMC4</span><span class="op">)</span></code></pre></div>
<p><img src="Sequencing_Effect_files/figure-html/PC_Selection_PBMC4-1.png" width="672" style="display: block; margin: auto;"></p>
</div>
<div id="generating-a-umap-and-clusters" class="section level3">
<h3 class="hasAnchor">
<a href="#generating-a-umap-and-clusters" class="anchor"></a>Generating a UMAP and clusters</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">PBMC4</span> <span class="op">&lt;-</span> <span class="va">PBMC4</span> <span class="op">%&gt;%</span> 
         <span class="fu">RunUMAP</span><span class="op">(</span>reduction <span class="op">=</span> <span class="st">"pca"</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">16</span><span class="op">)</span> <span class="op">%&gt;%</span>
         <span class="fu">FindNeighbors</span><span class="op">(</span>reduction <span class="op">=</span> <span class="st">"pca"</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">16</span><span class="op">)</span> <span class="op">%&gt;%</span>
         <span class="fu">FindClusters</span><span class="op">(</span>resolution <span class="op">=</span> <span class="fl">.8</span><span class="op">)</span></code></pre></div>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 12721
## Number of edges: 437773
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.8794
## Number of communities: 21
## Elapsed time: 1 seconds</code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">UMAPPlot</span><span class="op">(</span><span class="va">PBMC4</span>, cols <span class="op">=</span> <span class="va">colors.use</span>, group.by <span class="op">=</span> <span class="st">"orig.ident"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Default Workflow"</span><span class="op">)</span></code></pre></div>
<p><img src="Sequencing_Effect_files/figure-html/Clustering_PBMC4-1.png" width="672" style="display: block; margin: auto;"></p>
</div>
<div id="measuring-sample-umap-integration-generating-an-ilisi-score" class="section level3">
<h3 class="hasAnchor">
<a href="#measuring-sample-umap-integration-generating-an-ilisi-score" class="anchor"></a>Measuring sample-UMAP integration (generating an iLISI score)</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/GetiLISI.html">GetiLISI</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMC4</span>, nSamples <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0.03041182</code></pre>
</div>
</div>
<div id="cell-typing-of-joint-pbmcs-object-for-cms" class="section level2">
<h2 class="hasAnchor">
<a href="#cell-typing-of-joint-pbmcs-object-for-cms" class="anchor"></a>Cell Typing of joint PBMCs object for CMS</h2>
<div id="convert-cluster-classifications-to-cell-type-classifications" class="section level3">
<h3 class="hasAnchor">
<a href="#convert-cluster-classifications-to-cell-type-classifications" class="anchor"></a>Convert cluster classifications to cell type classifications</h3>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># For complete cell classification workflow see our vignette "Biaxial Gating of a Single Sample"</span>
<span class="co"># More details can be found in figure S3 of our manuscript "Data Matrix Normalization and Merging Strategies Minimize Batch-specific Systemic Variation in scRNA-Seq Data."</span>
<span class="fu">UMAPPlot</span><span class="op">(</span><span class="va">PBMC4</span>, cols <span class="op">=</span> <span class="va">colors.use</span>, pt.size <span class="op">=</span> <span class="fl">2</span>, 
         group.by <span class="op">=</span> <span class="st">"seurat_clusters"</span>, label <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></code></pre></div>
<p><img src="Sequencing_Effect_files/figure-html/Cluster_assignment_PBMC4-1.png" width="672" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># B_Naive = 7</span>
<span class="co"># B_Memory = 12, 18</span>
<span class="co"># T_CD4 = 0, 1, 4, 6, 16</span>
<span class="co"># TReg = 11</span>
<span class="co"># T_CD8 = 8, 13</span>
<span class="co"># NK_T = 5, 15</span>
<span class="co"># NK = 2, 3</span>
<span class="co"># NK_CD56Hi = 9</span>
<span class="co"># Monocyte_Classical = 10</span>
<span class="co"># Monocyte_NonClassical = 14</span>
<span class="co"># Dendritic_Cells = 17</span>
<span class="co"># HSPCs = 20</span>
<span class="co"># Cycling_Cells = 19</span>
<span class="fu">Idents</span><span class="op">(</span><span class="va">PBMC4</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">PBMC4</span><span class="op">[[</span><span class="st">"seurat_clusters"</span><span class="op">]</span><span class="op">]</span>
<span class="fu">Idents</span><span class="op">(</span><span class="va">PBMC4</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">plyr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyr/man/mapvalues.html">mapvalues</a></span><span class="op">(</span><span class="fu">Idents</span><span class="op">(</span><span class="va">PBMC4</span><span class="op">)</span>, from <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">7</span>, <span class="fl">12</span>, <span class="fl">18</span>, <span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">4</span>, <span class="fl">6</span>, <span class="fl">16</span>,
                                                         <span class="fl">11</span>, <span class="fl">8</span>, <span class="fl">13</span>, <span class="fl">5</span>, <span class="fl">15</span>, <span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">9</span>,
                                                         <span class="fl">10</span>, <span class="fl">14</span>, <span class="fl">17</span>,
                                                         <span class="fl">20</span>, <span class="fl">19</span><span class="op">)</span>, 
                to <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'B_Naive'</span>, <span class="st">'B_Memory'</span>, <span class="st">'B_Memory'</span>, <span class="st">'T_CD4'</span>, <span class="st">'T_CD4'</span>, <span class="st">'T_CD4'</span>, <span class="st">'T_CD4'</span>, <span class="st">'T_CD4'</span>,
                       <span class="st">'TReg'</span>, <span class="st">'T_CD8'</span>, <span class="st">'T_CD8'</span>, <span class="st">'NK_T'</span>, <span class="st">'NK_T'</span>, <span class="st">'NK'</span>, <span class="st">'NK'</span>, <span class="st">'NK_CD56Hi'</span>,
                       <span class="st">'Monocyte_Classical'</span>, <span class="st">'Monocyte_NonClassical'</span>, <span class="st">'Dendritic_Cells'</span>,
                       <span class="st">'HSPCs'</span>, <span class="st">'Cycling_Cells'</span><span class="op">)</span><span class="op">)</span>
<span class="fu">Idents</span><span class="op">(</span><span class="va">PBMC4</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="fu">Idents</span><span class="op">(</span><span class="va">PBMC4</span><span class="op">)</span>,
                        levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"B_Naive"</span>, <span class="st">"B_Memory"</span>, <span class="st">"T_CD4"</span>, <span class="st">"TReg"</span>,
                                   <span class="st">"T_CD8"</span>, <span class="st">"NK_T"</span>, <span class="st">"NK"</span>, <span class="st">"NK_CD56Hi"</span>,
                                   <span class="st">"Monocyte_Classical"</span>, <span class="st">"Monocyte_NonClassical"</span>,
                                   <span class="st">"Dendritic_Cells"</span>, <span class="st">"HSPCs"</span>, <span class="st">"Cycling_Cells"</span><span class="op">)</span><span class="op">)</span>
<span class="va">PBMC4</span><span class="op">[[</span><span class="st">"Cell_Type"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">Idents</span><span class="op">(</span><span class="va">PBMC4</span><span class="op">)</span>
<span class="fu">UMAPPlot</span><span class="op">(</span><span class="va">PBMC4</span>, cols <span class="op">=</span> <span class="va">colors.use</span>, label <span class="op">=</span> <span class="cn">F</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Cell Type Classifications"</span><span class="op">)</span></code></pre></div>
<p><img src="Sequencing_Effect_files/figure-html/Cluster_assignment_PBMC4-2.png" width="672" style="display: block; margin: auto;"></p>
</div>
<div id="compare-single-sample-workflow-cell-type-classifications-to-joint-classifications-generating-a-cms" class="section level3">
<h3 class="hasAnchor">
<a href="#compare-single-sample-workflow-cell-type-classifications-to-joint-classifications-generating-a-cms" class="anchor"></a>Compare single-sample workflow cell type classifications to joint classifications (generating a CMS)</h3>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># PBMC Sample 4-A</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">'BatchNorm'</span>, <span class="va">PBMC4A_Single_ID</span><span class="op">)</span>
<span class="va">S4Acms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetCMS.html">GetCMS</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMC4</span>, 
                sample.ID <span class="op">=</span> <span class="st">"Sample_4A"</span>, 
                reference.ID <span class="op">=</span> <span class="va">PBMC4A_Single_ID</span><span class="op">)</span>
<span class="co"># PBMC Sample 4-B</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">'BatchNorm'</span>, <span class="va">PBMC4B_Single_ID</span><span class="op">)</span>
<span class="va">S4Bcms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetCMS.html">GetCMS</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMC4</span>, 
                sample.ID <span class="op">=</span> <span class="st">"Sample_4B"</span>, 
                reference.ID <span class="op">=</span> <span class="va">PBMC4B_Single_ID</span><span class="op">)</span>

<span class="co"># Average CMS</span>
<span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">S4Acms</span>, <span class="va">S4Bcms</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0.03208177</code></pre>
</div>
</div>
</div>
<div id="non-identical-duplicate-differing-depths-sequencing" class="section level1">
<h1 class="hasAnchor">
<a href="#non-identical-duplicate-differing-depths-sequencing" class="anchor"></a>Non-identical (duplicate, differing depths) Sequencing</h1>
<div id="processing-the-seurat-object-1" class="section level2">
<h2 class="hasAnchor">
<a href="#processing-the-seurat-object-1" class="anchor"></a>Processing the Seurat Object</h2>
<div id="data-filtering-normalization-1" class="section level3">
<h3 class="hasAnchor">
<a href="#data-filtering-normalization-1" class="anchor"></a>Data filtering &amp; normalization</h3>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Import unfiltered Seurat object (included with 'BatchNorm' package)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">'BatchNorm'</span>, <span class="va">PBMC5</span><span class="op">)</span>

<span class="co"># Run "standard" Seurat workflow: </span>
<span class="co"># Including filtering by mitochondrial percentage (+5 SD)</span>
<span class="co"># Including data normalization, variable gene selection and gene scaling (performed on all samples together)</span>
<span class="va">PBMC5</span> <span class="op">&lt;-</span> <span class="va">PBMC5</span> <span class="op">%&gt;%</span> 
         <span class="fu"><a href="../reference/MitoFilter.html">MitoFilter</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> 
         <span class="fu">NormalizeData</span><span class="op">(</span>normalization.method <span class="op">=</span> <span class="st">"LogNormalize"</span>, assay <span class="op">=</span> <span class="st">"RNA"</span>, scale.factor <span class="op">=</span> <span class="fl">10000</span><span class="op">)</span> <span class="op">%&gt;%</span> 
         <span class="fu">NormalizeData</span><span class="op">(</span>verbose <span class="op">=</span> <span class="cn">FALSE</span>, assay <span class="op">=</span> <span class="st">"ADT"</span>, normalization.method <span class="op">=</span> <span class="st">"CLR"</span><span class="op">)</span> <span class="op">%&gt;%</span>
         <span class="fu">FindVariableFeatures</span><span class="op">(</span>selection.method <span class="op">=</span> <span class="st">"vst"</span>, nfeatures <span class="op">=</span> <span class="fl">2000</span><span class="op">)</span> <span class="op">%&gt;%</span>
         <span class="fu">ScaleData</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> 
         <span class="fu">RunPCA</span><span class="op">(</span>npcs <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></code></pre></div>
</div>
<div id="selecting-the-appropriate-number-of-principal-components-for-umap-reduction-1" class="section level3">
<h3 class="hasAnchor">
<a href="#selecting-the-appropriate-number-of-principal-components-for-umap-reduction-1" class="anchor"></a>Selecting the appropriate number of Principal Components for UMAP reduction</h3>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Identify correct numbers of PCs</span>
<span class="va">PBMC5.pca.test</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/TestPCA.html">TestPCA</a></span><span class="op">(</span><span class="va">PBMC5</span><span class="op">)</span>
<span class="va">PBMC5.pca.test</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">20</span><span class="op">]</span></code></pre></div>
<pre><code>##                   [,1]        [,2]       [,3]       [,4]       [,5]        [,6]
## SD         10.58363440  6.72990013 5.33404735 4.18319372 4.00809819 3.201402719
## Proportion  0.08951584  0.03619491 0.02273756 0.01398448 0.01283828 0.008190508
## Cumulative  0.08951584  0.12571074 0.14844831 0.16243278 0.17527106 0.183461573
## ZScore     38.63602574 15.49285345 9.65189068 5.85273976 5.35525150 3.337952812
##                   [,7]        [,8]        [,9]       [,10]       [,11]
## SD         3.050269386 2.580910878 2.391094406 2.275051716 2.265910779
## Proportion 0.007435439 0.005323242 0.004569026 0.004136307 0.004103135
## Cumulative 0.190897011 0.196220254 0.200789280 0.204925587 0.209028722
## ZScore     3.010226103 2.093457970 1.766101548 1.578286015 1.563888285
##                  [,12]       [,13]       [,14]       [,15]       [,16]
## SD         2.155599448 1.993519631 1.886166179 1.840696473 1.816369254
## Proportion 0.003713354 0.003175932 0.002843087 0.002707663 0.002636565
## Cumulative 0.212742076 0.215918008 0.218761095 0.221468758 0.224105323
## ZScore     1.394709406 1.161449432 1.016982665 0.958203837 0.927344925
##                  [,17]      [,18]       [,19]       [,20]
## SD         1.767894224 1.75277289 1.713076660 1.674299850
## Proportion 0.002497714 0.00245517 0.002345221 0.002240251
## Cumulative 0.226603038 0.22905821 0.231403428 0.233643679
## ZScore     0.867078782 0.84861289 0.800891354 0.755330596</code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># 14 PCs with z &gt; 1</span>
<span class="co"># Proceed with 14 PCs for dimensional reduction &amp; clustering</span>
<span class="co"># Visualize PCs plotted by standard deviation:</span>
<span class="fu">ElbowPlot</span><span class="op">(</span><span class="va">PBMC5</span><span class="op">)</span></code></pre></div>
<p><img src="Sequencing_Effect_files/figure-html/PC_Selection_PBMC5-1.png" width="672" style="display: block; margin: auto;"></p>
</div>
<div id="generating-a-umap-and-clusters-1" class="section level3">
<h3 class="hasAnchor">
<a href="#generating-a-umap-and-clusters-1" class="anchor"></a>Generating a UMAP and clusters</h3>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">PBMC5</span> <span class="op">&lt;-</span> <span class="va">PBMC5</span> <span class="op">%&gt;%</span> 
         <span class="fu">RunUMAP</span><span class="op">(</span>reduction <span class="op">=</span> <span class="st">"pca"</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">14</span><span class="op">)</span> <span class="op">%&gt;%</span>
         <span class="fu">FindNeighbors</span><span class="op">(</span>reduction <span class="op">=</span> <span class="st">"pca"</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">14</span><span class="op">)</span> <span class="op">%&gt;%</span>
         <span class="fu">FindClusters</span><span class="op">(</span>resolution <span class="op">=</span> <span class="fl">.8</span><span class="op">)</span></code></pre></div>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 7764
## Number of edges: 270395
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.8881
## Number of communities: 19
## Elapsed time: 0 seconds</code></pre>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">UMAPPlot</span><span class="op">(</span><span class="va">PBMC5</span>, cols <span class="op">=</span> <span class="va">colors.use</span>, group.by <span class="op">=</span> <span class="st">"orig.ident"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Default Workflow"</span><span class="op">)</span></code></pre></div>
<p><img src="Sequencing_Effect_files/figure-html/Clustering_PBMC5-1.png" width="672" style="display: block; margin: auto;"></p>
</div>
<div id="measuring-sample-umap-integration-generating-an-ilisi-score-1" class="section level3">
<h3 class="hasAnchor">
<a href="#measuring-sample-umap-integration-generating-an-ilisi-score-1" class="anchor"></a>Measuring sample-UMAP integration (generating an iLISI score)</h3>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/GetiLISI.html">GetiLISI</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMC5</span>, nSamples <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0.0275641</code></pre>
</div>
</div>
<div id="cell-typing-of-joint-pbmcs-object-for-cms-1" class="section level2">
<h2 class="hasAnchor">
<a href="#cell-typing-of-joint-pbmcs-object-for-cms-1" class="anchor"></a>Cell Typing of joint PBMCs object for CMS</h2>
<div id="convert-cluster-classifications-to-cell-type-classifications-1" class="section level3">
<h3 class="hasAnchor">
<a href="#convert-cluster-classifications-to-cell-type-classifications-1" class="anchor"></a>Convert cluster classifications to cell type classifications</h3>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># For complete cell classification workflow see our vignette "Biaxial Gating of a Single Sample"</span>
<span class="co"># More details can be found in figure S3 of our manuscript "Data Matrix Normalization and Merging Strategies Minimize Batch-specific Systemic Variation in scRNA-Seq Data."</span>
<span class="fu">UMAPPlot</span><span class="op">(</span><span class="va">PBMC5</span>, cols <span class="op">=</span> <span class="va">colors.use</span>, pt.size <span class="op">=</span> <span class="fl">2</span>, 
         group.by <span class="op">=</span> <span class="st">"seurat_clusters"</span>, label <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></code></pre></div>
<p><img src="Sequencing_Effect_files/figure-html/Cluster_assignment_PBMC5-1.png" width="672" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># B_Naive = 11</span>
<span class="co"># B_Memory = 12</span>
<span class="co"># T_CD4 = 0, 1, 2, 3, 16</span>
<span class="co"># TReg not detected</span>
<span class="co"># T_CD8 = 6</span>
<span class="co"># NK_T = 4, 7, 8, 10</span>
<span class="co"># NK = 9</span>
<span class="co"># NK_CD56Hi = 9</span>
<span class="co"># Monocyte_Classical = 5, 15</span>
<span class="co"># Monocyte_NonClassical = 14</span>
<span class="co"># Dendritic_Cells = 17, 18</span>
<span class="co"># HSPCs = 13 (contains erythrocytes)</span>
<span class="fu">Idents</span><span class="op">(</span><span class="va">PBMC5</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">PBMC5</span><span class="op">[[</span><span class="st">"seurat_clusters"</span><span class="op">]</span><span class="op">]</span>
<span class="fu">Idents</span><span class="op">(</span><span class="va">PBMC5</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">plyr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyr/man/mapvalues.html">mapvalues</a></span><span class="op">(</span><span class="fu">Idents</span><span class="op">(</span><span class="va">PBMC5</span><span class="op">)</span>, from <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">11</span>, <span class="fl">12</span>, <span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">16</span>,
                                                         <span class="fl">6</span>, <span class="fl">4</span>, <span class="fl">7</span>, <span class="fl">8</span>, <span class="fl">10</span>, <span class="fl">9</span>,
                                                         <span class="fl">5</span>, <span class="fl">15</span>, <span class="fl">14</span>,
                                                         <span class="fl">17</span>, <span class="fl">18</span>, <span class="fl">13</span><span class="op">)</span>, 
                to <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'B_Naive'</span>, <span class="st">'B_Memory'</span>, <span class="st">'T_CD4'</span>, <span class="st">'T_CD4'</span>, <span class="st">'T_CD4'</span>, <span class="st">'T_CD4'</span>, <span class="st">'T_CD4'</span>,
                       <span class="st">'T_CD8'</span>, <span class="st">'NK_T'</span>, <span class="st">'NK_T'</span>, <span class="st">'NK_T'</span>, <span class="st">'NK_T'</span>, <span class="st">'NK'</span>,
                       <span class="st">'Monocyte_Classical'</span>, <span class="st">'Monocyte_Classical'</span>, <span class="st">'Monocyte_NonClassical'</span>,
                       <span class="st">'Dendritic_Cells'</span>, <span class="st">'Dendritic_Cells'</span>, <span class="st">'HSPCs'</span><span class="op">)</span><span class="op">)</span>
<span class="fu">Idents</span><span class="op">(</span><span class="va">PBMC5</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="fu">Idents</span><span class="op">(</span><span class="va">PBMC5</span><span class="op">)</span>,
                        levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"B_Naive"</span>, <span class="st">"B_Memory"</span>, <span class="st">"T_CD4"</span>, <span class="st">"TReg"</span>,
                                   <span class="st">"T_CD8"</span>, <span class="st">"NK_T"</span>, <span class="st">"NK"</span>, <span class="st">"NK_CD56Hi"</span>,
                                   <span class="st">"Monocyte_Classical"</span>, <span class="st">"Monocyte_NonClassical"</span>,
                                   <span class="st">"Dendritic_Cells"</span>, <span class="st">"HSPCs"</span>, <span class="st">"Cycling_Cells"</span><span class="op">)</span><span class="op">)</span>
<span class="va">PBMC5</span><span class="op">[[</span><span class="st">"Cell_Type"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">Idents</span><span class="op">(</span><span class="va">PBMC5</span><span class="op">)</span>
<span class="fu">UMAPPlot</span><span class="op">(</span><span class="va">PBMC5</span>, cols <span class="op">=</span> <span class="va">colors.use</span>, label <span class="op">=</span> <span class="cn">F</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Cell Type Classifications"</span><span class="op">)</span></code></pre></div>
<p><img src="Sequencing_Effect_files/figure-html/Cluster_assignment_PBMC5-2.png" width="672" style="display: block; margin: auto;"></p>
</div>
<div id="compare-single-sample-workflow-cell-type-classifications-to-joint-classifications-generating-a-cms-1" class="section level3">
<h3 class="hasAnchor">
<a href="#compare-single-sample-workflow-cell-type-classifications-to-joint-classifications-generating-a-cms-1" class="anchor"></a>Compare single-sample workflow cell type classifications to joint classifications (generating a CMS)</h3>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># PBMC Sample 5-A</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">'BatchNorm'</span>, <span class="va">PBMC5A_Single_ID</span><span class="op">)</span>
<span class="va">S5Acms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetCMS.html">GetCMS</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMC5</span>, 
                sample.ID <span class="op">=</span> <span class="st">"Sample_5A"</span>, 
                reference.ID <span class="op">=</span> <span class="va">PBMC5A_Single_ID</span><span class="op">)</span>
<span class="co"># PBMC Sample 5-B</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">'BatchNorm'</span>, <span class="va">PBMC5B_Single_ID</span><span class="op">)</span>
<span class="va">S5Bcms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetCMS.html">GetCMS</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">PBMC5</span>, 
                sample.ID <span class="op">=</span> <span class="st">"Sample_5B"</span>, 
                reference.ID <span class="op">=</span> <span class="va">PBMC5B_Single_ID</span><span class="op">)</span>

<span class="co"># Average CMS</span>
<span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">S5Acms</span>, <span class="va">S5Bcms</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0.03288283</code></pre>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Benjamin Babcock.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
